<!DOCTYPE html>
<!-- saved from url=(0040)https://secondary-structure.netlify.app/ -->
<html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Secondary Structure</title>

<link rel="stylesheet" href="./style.css">

<link rel="shortcut icon" href="https://secondary-structure.netlify.app/#">

<script src="https://gjmol.org/jsmol/JSmol.min.js" type="text/javascript"></script>

<script type="text/javascript">
function getValueByKey(object, row) {
  return object[row];
}
const moleculeProps = {
    "idealalpha": { "type": "helices", "chains": [["A", "A"]], "residues": [[2, 22]] },
    "2TRX": { "type": "strands", "chains": [["A", "A"], ["A", "A"], ["A", "A"], ["A", "A"], ["A", "A"]], "residues": [[3, 8], [21, 28], [53, 59], [76, 82], [86, 92]] },
    "1ARB": { "type": "turns", "chains": [["A", "A"]], "residues": [[109, 114]] },
    "6EC0": { "type": "regular", "chains": [["A", "A"]], "residues": [[225, 331]] },
    "1U4G": { "type": "distorted", "chains": [["A", "A"]], "residues": [[134, 149]], "caption": "Natural Œ±-helix with small irregularities" },
    "1V54": { "type": "distorted", "chains": [["E", "E"]], "residues": [[80, 97]], "caption": "A kink in an Œ±-helix" },
    "1QOY": { "type": "distorted", "chains": [["A", "A"]], "residues": [[18, 36]], "caption": "Distortions of an Œ±-helix leading to a looser C-terminus" },
    "2CPP": { "type": "distorted", "chains": [["A", "A"]], "residues": [[258, 276]], "caption": "A strong kink due to a Pro residue in a slightly distorted Œ±-helix" },
    "1TVF": { "type": "distorted", "chains": [["A", "A"]], "residues": [[73, 91]], "caption": "Difference between body and N-terminus of an Œ±-helix" },
    "5MBN": { "type": "distorted", "chains": [["A", "A"]], "residues": [[3, 19]], "caption": "An Œ±<sub>œÄ</sub> cap at the C-terminus of an Œ±-helix leading to higher curvature" },
    "1QTE": { "type": "distorted", "chains": [["A", "A"]], "residues": [[21, 36]], "caption": "Transition from a helix to a turn region with gradually increasing curvatures and interspersed high torsions" },
    "1RWZ": { "type": "distorted", "chains": [["A", "A"]], "residues": [[8, 23]], "caption": "C-terminal caps of Œ±-helix" },
    "1KSS": { "type": "distorted", "chains": [["A", "A"]], "residues": [[543, 556]], "caption": "N-terminal caps of Œ±-helix" },
    "1GPB": { "type": "310", "chains": [["A", "A"]], "residues": [[515, 525]] },
    "2QD1": { "type": "pi", "chains": [["A", "A"]], "residues": [[341, 361]] },
    "2BEG": { "type": "parallel", "chains": [["B", "B"], ["C", "C"], ["D", "D"]], "residues": [[18, 26], [18, 26], [18, 26]] },
    "1OFS": { "type": "lefttwist", "chains": [["A", "A"], ["B", "B"]], "residues": [[2, 8], [39, 46]] },
    "1BA7": { "type": "righttwist", "chains": [["A", "A"], ["A", "A"]], "residues": [[94, 96], [104, 106]] },
    "2YXF": { "type": "relaxed", "chains": [["A", "A"], ["A", "A"]], "residues": [[51, 56], [62, 69]] },
    "3CNA": { "type": "bulge", "chains": [["A", "A"], ["A", "A"]], "residues": [[35, 36], [27, 27]], "subtype": ["Classic bulge"], "caption": ["Bulge description..."] },
    "4CHA": { "type": "bulge", "chains": [["C", "C"], ["C", "C"]], "residues": [[196, 197], [213, 213]], "subtype": ["G1 Œ≤-bulge"] },
    "5CPA": { "type": "bulge", "chains": [["A", "A"], ["A", "A"], ["A", "A"], ["A", "A"]], "residues": [[130, 131], [140, 140], [199, 200], [271, 271]], "subtype": ["Gx NH Œ≤-bulge", "Gx CO bulge"] },
    "155C": { "type": "bulge", "chains": [["A", "A"], ["A", "A"]], "residues": [[35, 36], [19, 19]], "subtype": ["Parallel bulge"] },
    "6LUX": { "type": "bulge", "chains": [["A", "A"], ["A", "A"]], "residues": [[46, 47], [81, 81]], "subtype": ["Wide bulge"] },
    "2BK9": { "type": "betaturn", "chains": [["A", "A"]], "residues": [[147, 150]], "subtype": ["Type I"] },
    "1KKO": { "type": "betaturn", "chains": [["A", "A"]], "residues": [[77, 80]], "subtype": ["Type I'"] },
    "1H16": { "type": "betaturn", "chains": [["A", "A"]], "residues": [[424, 427]], "subtype": ["Type II"] },
    "1UXA": { "type": "betaturn", "chains": [["A", "A"]], "residues": [[213, 216]], "subtype": ["Type II'"] },
    "1JYK": { "type": "betaturn", "chains": [["A", "A"]], "residues": [[32, 35]], "subtype": ["Type IV1"] },
    "1URS": { "type": "betaturn", "chains": [["A", "A"]], "residues": [[296, 299]], "subtype": ["Type IV2"] },
    "1PA7": { "type": "betaturn", "chains": [["A", "A"]], "residues": [[125, 128]], "subtype": ["Type IV3"] },
    "1QWG": { "type": "betaturn", "chains": [["A", "A"]], "residues": [[194, 197]], "subtype": ["Type IV4"] },
    "1HBN": { "type": "betaturn", "chains": [["E", "E"]], "residues": [[163, 166]], "subtype": ["Type VIa1"] },
    "1IQ6": { "type": "betaturn", "chains": [["A", "A"]], "residues": [[68, 71]], "subtype": ["Type VIa2"] },
    "1YT3": { "type": "betaturn", "chains": [["A", "A"]], "residues": [[141, 144]], "subtype": ["Type VIb"] },
    "1SU8": { "type": "betaturn", "chains": [["A", "A"]], "residues": [[435, 438]], "subtype": ["Type VIII"] },
    "1AAP": { "type": "alphaturn", "chains": [["B", "B"]], "residues": [[24, 28]], "subtype": ["Type I-Œ±RS"] },
    "1FCB": { "type": "alphaturn", "chains": [["B", "B"]], "residues": [[372, 376]], "subtype": ["Type I-Œ±LS"] },
    "3COX": { "type": "alphaturn", "chains": [["A", "A"]], "residues": [[453, 457]], "subtype": ["Type II-Œ±RS"] },
    "1L76": { "type": "alphaturn", "chains": [["A", "A"]], "residues": [[27, 31]], "subtype": ["Type II-Œ±LS"] },
    "4GCR": { "type": "alphaturn", "chains": [["A", "A"]], "residues": [[136, 140]], "subtype": ["Type I-Œ±LU"] },
    "1BBP": { "type": "alphaturn", "chains": [["C", "C"]], "residues": [[142, 146]], "subtype": ["Type I-Œ±RU"] },
    "1GP1": { "type": "alphaturn", "chains": [["B", "B"]], "residues": [[75, 79]], "subtype": ["Type II-Œ±LU"] },
    "1CMS": { "type": "alphaturn", "chains": [["A", "A"]], "residues": [[11, 15]], "subtype": ["Type II-Œ±RU"] },
    "1S01": { "type": "alphaturn", "chains": [["A", "A"]], "residues": [[166, 170]], "subtype": ["Type I-Œ±C"] },
    "2AYH": { "type": "piturn", "chains": [["A", "A"]], "residues": [[96, 101]], "subtype": ["Type œÄŒ±L"] },
    "4ENL": { "type": "piturn", "chains": [["A", "A"]], "residues": [[67, 73]], "subtype": ["Type œÄŒ±R"] },
    "1HMY": { "type": "piturn", "chains": [["A", "A"]], "residues": [[127, 132]], "subtype": ["Type œÄ'Œ±L"] },
    "1RND": { "type": "piturn", "chains": [["A", "A"]], "residues": [[111, 116]], "subtype": ["Type œÄŒ≤"] },
    "1E1H": { "type": "piturn", "chains": [["B", "B"]], "residues": [[324, 329]], "subtype": ["Type œÄ-NHB"] },
    "1AJS": { "type": "piturn", "chains": [["A", "A"]], "residues": [[162, 167]], "subtype": ["Type œÄ-HB, isolated"] },
    "16PK": { "type": "piturn", "chains": [["A", "A"]], "residues": [[382, 387]], "subtype": ["Type œÄ-HB SCH+Œ≤"] },
    "8TLN": { "type": "gammaturn", "chains": [["E", "E"]], "residues": [[25, 27]], "subtype": ["Classic"] },
    "2SGA": { "type": "gammaturn", "chains": [["A", "A"]], "residues": [[113, 115]], "subtype": ["Inverse"] },
    "1NJH": { "type": "deltaturn", "chains": [["A", "A"]], "residues": [[63, 64]], "subtype": ["Gly-Pro"] },
    "2BMH": { "type": "deltaturn", "chains": [["A", "A"]], "residues": [[195, 196]], "subtype": ["Asp-Pro"] },
    "1KB0": { "type": "epsilonturn", "chains": [["A", "A"]], "residues": [[617, 619]], "subtype": ["1"] },
    "1QXY": { "type": "epsilonturn", "chains": [["A", "A"]], "residues": [[171, 173]], "subtype": ["2"] },
    "3O12": { "type": "epsilonturn", "chains": [["A", "A"]], "residues": [[110, 112]], "subtype": ["3"] },
    "2RA6": { "type": "epsilonturn", "chains": [["A", "A"]], "residues": [[93, 95]], "subtype": ["4"] },
    "3F9M": { "type": "epsilonturn", "chains": [["A", "A"]], "residues": [[227, 229]], "subtype": ["5"] },
    "1HPG": { "type": "epsilonturn", "chains": [["A", "A"]], "residues": [[221, 223]], "subtype": ["6"] },
    "1LRI": { "type": "coil", "chains": [["A", "A"], ["A", "A"], ["A", "A"]], "residues": [[1, 3], [33, 42], [67, 72]] }
};
 var JmolInfo = {
 width:'100%',
 height:'100%',
 color:'#FFFFFF',
 j2sPath: 'https://chemapps.stolaf.edu/jmol/jsmol/j2s',// Points to remote library
 use:'html5',
 loadStructCallback:'moleculeLoaded',
 measureCallback:'measurementDone',
 messageCallback:'messageReceived'
}

const currentIDs = {
  "bulge": 0,
  "piturn": 0,
  "alphaturn": 0,
  "betaturn": 0,
  "gammaturn": 0,
  "deltaturn": 0,
  "epsilonturn": 0
}

const currentMeasurementLabels = {
  "piturn": 0,
  "alphaturn": 0,
  "betaturn": 0,
  "gammaturn": 0,
  "deltaturn": 0,
  "epsilonturn": 0
}

var contentLoaded = {
"secondarystructures": false,
"helices": false,
"strands": false,
"turns": false,
"regular": false,
"distorted": false,
"threeten": false,
"pihelix": false,
"parallel": false,
"antiparallel": false,
"lefttwist": false,
"relaxed": false,
"righttwist": false,
"bulge": false,
"piturns": false,
"alphaturns": false,
"betaturns": false,
"gammaturns": false,
"epsilonturns": false,
"deltaturns": false,
"coil": false
}

var jmol_intro = 'jmol_intro';
var jmol_helices = 'jmol_helices';
var jmol_strands = 'jmol_strands';
var jmol_turns = 'jmol_turns';
var jmol_regular = 'jmol_regular';
var jmol_distorted = 'jmol_distorted';
var jmol_310 = 'jmol_310';
var jmol_pi = 'jmol_pi';
var jmol_parallel = 'jmol_parallel';
var jmol_lefttwist = 'jmol_lefttwist';
var jmol_relaxed = 'jmol_relaxed';
var jmol_righttwist = 'jmol_righttwist';
var jmol_bulge = 'jmol_bulge';
var jmol_piturn = 'jmol_piturn';
var jmol_alphaturn = 'jmol_alphaturn';
var jmol_betaturn = 'jmol_betaturn';
var jmol_gammaturn = 'jmol_gammaturn';
var jmol_deltaturn = 'jmol_deltaturn';
var jmol_epsilonturn = 'jmol_epsilonturn';
var jmol_coil = 'jmol_coil';
 
var typingTimer;
var doneTypingInterval = 500;
var sequenceChanges = [];

var dragState = 0;
var introDihedralAngles = [];
var scriptBusy = false;

function messageReceived() {
	let message = arguments[1];
	//console.log("JSMol: " + message);
	if (message.startsWith("Measurement Information")) {
		//console.log("Measurement Information obtained.");
		introDihedralAngles = message.split(/\r?\n|\r|\n/g);
		introDihedralAngles.forEach((item, index, arr) => {
			introDihedralAngles[index] = item.split("\t");
		});
		dragState = 1;
	}
	if (message == "Script completed") {
		//console.log("JSMol: " + message);
		scriptBusy = false;
	}
}

$(document).ready(function(){	
	// Restrict input to one letter codes for amino acids
	const regex = new RegExp("^[acdefghiklmnpqrstvwy]*$");

	document.getElementById("sequence").addEventListener("beforeinput", (event) => {
	  if (event.data != null && !regex.test(event.data)) 
	    event.preventDefault();
	});
	
	// Monitor key strokes in the sequence
	
	document.getElementById("sequence").onkeyup = function(evt) {
		clearTimeout(typingTimer);
		typingTimer = setTimeout(commitSequenceChanges, doneTypingInterval);
	};
	
	document.getElementById("sequence").onkeydown = function(evt) {
		clearTimeout(typingTimer);
		const allowedKeyCodes = [8, 46, 65, 67, 68, 69, 70, 71, 72, 73, 75, 76, 77, 78, 80, 81, 82, 83, 84, 86, 87, 89]; // Only look for Backspace, Delete, and the 20 single letter amino acid codes
		if (allowedKeyCodes.includes(evt.keyCode)) {
			storeSequenceChanges(evt);
		}
	};
	
	// Monitor dragging in sequence input field
	
	var dragging = false;
	
	document.getElementById("sequence").ondragstart = function(evt) {
		if (dragging == false) {
			dragging = true;
			storeSequenceChanges(evt);
		}
	};

	document.getElementById("sequence").ondragend = function(evt) {
		dragging = false;
		Jmol.script(jmol_intro, 'script preLoadingScript.txt;');
		storeSequenceChanges(evt);
		commitSequenceChanges();
	};
	
	// Show molecule for default sequence

	mainTextControl();

	$('#JmolDiv_intro').html( Jmol.getAppletHtml(jmol_intro, JmolInfo) );
	Jmol.script(jmol_intro, 'set disablePopupMenu FALSE;frank off;set perspectiveDepth off;selectionHalos ON;color SELECTIONHALOS [103, 203, 223];set showHiddenSelectionHalos FALSE;');

	sequenceInitialise();
	
	updateRamachandran();
});
</script>

<script type="text/javascript">
 Jmol.setLinkCss(null, "style='color:#67cbdf'");
</script>

</head>
<body>
	
<div class="mainBox">
	<div class="loading" style="visibility: hidden;">
	  <img id="loading-image" src="./cursor_wait.gif" alt="Loading...">
	</div>
	
	<div id="secondarystructures" class="structureBox clickeable">
		<div id="sselements_title" class="structureBoxTitle" style="pointer-events:none;"><b>Secondary Structure Elements</b></div>
			<div id="helices" class="structureBox clickeable" style="display:block;align-items:center;">
				<div id="helices_title" class="structureBoxTitle" style="pointer-events:none;">Helices</div>
				<div id="pihelix" class="structureBox singleItem clickeable">
					<div style="pointer-events:none;">œÄ‚ÄëHelices&nbsp;<span style="color:rgba(255, 255, 255, 0.5);">(<i>i</i> ‚Üí <i>i</i> - 5)</span></div>
				</div>
				<div id="alphahelices" class="structureBox" style="display:block;">
					<div id="alphahelices_title" class="structureBoxTitle">Œ±-Helices&nbsp;<span style="color:rgba(255, 255, 255, 0.5);">(<i>i</i> ‚Üí <i>i</i> - 4)</span></div>
					<div id="regular" class="structureBox singleItem clickeable">
						Regular
					</div>
					<div id="distorted" class="structureBox singleItem clickeable">
						Distorted
					</div>
				</div>
				<div id="threeten" class="structureBox singleItem clickeable">
					<div style="pointer-events:none;">3<sub>10</sub>‚ÄëHelices&nbsp;<span style="color:rgba(255, 255, 255, 0.5);">(<i>i</i> ‚Üí <i>i</i> - 3)</span></div>
				</div>
			</div>
			<div id="strands" class="structureBox clickeable" style="position:relative;">
				<div id="strands_title" class="structureBoxTitle" style="pointer-events:none;">Œ≤-Strands</div>
				<div id="parallel" class="structureBox clickeable">
					Parallel
				</div>
				<div id="antiparallel" class="structureBox" style="border-style:dashed;">
					Anti-Parallel
					<div id="relaxed" class="structureBox clickeable">
						Relaxed
					</div>
					<div id="righttwist" class="structureBox clickeable">
						Right-Twisted
					</div>
					<div id="lefttwist" class="structureBox clickeable">
						Left-Twisted
					</div>
				</div>
				<div id="bulge" class="structureBox clickeable" style="position:absolute;top:105px;right:10px;transform: translate(0%, -50%);">
					<div id="bulge_title" class="structureBoxTitle" style="pointer-events:none;">Œ≤-Bulge</div>
					<div id="bulge_parallel" class="structureBox singleItem" style="pointer-events:none;">
						Parallel
					</div>
					<div id="bulge_classic" class="structureBox singleItem" style="pointer-events:none;">
						Classic
					</div>
					<div id="bulge_g1" class="structureBox singleItem" style="pointer-events:none;">
						G1
					</div>
					<div id="bulge_gx" class="structureBox singleItem" style="pointer-events:none;">
						Gx
					</div>
					<div id="bulge_wide" class="structureBox singleItem" style="pointer-events:none;">
						Wide
					</div>
				</div>
			</div>
			<div id="turns" class="structureBox clickeable">
				<div id="turns_title" class="structureBoxTitle" style="pointer-events:none;">Turns</div>
				<div id="piturns" class="structureBox">
					<div style="text-align:center;pointer-events:none;">œÄ-Turns<br><span style="color:rgba(255, 255, 255, 0.5);">(<i>n</i>‚Äâ=‚Äâ6)</span></div>
					<ul class="bulletList">
						<li>œÄ<sub>Œ±L</sub></li>
						<li>œÄ<sub>Œ±R</sub></li>
						<li>œÄ‚Äô<sub>Œ±L</sub></li>
						<li>œÄ<sub>Œ≤</sub></li>
						<br>
						<li>œÄ‚ÄëNHB</li>
						<li>œÄ‚ÄëHB</li>
						<li>SCH<sub>+Œ≤</sub></li>
						<li>SCH<sub>-Œ≤</sub></li>
					</ul>
				</div>
				<div id="alphaturns" class="structureBox">
					<div style="text-align:center;pointer-events:none;">Œ±-Turns<br><span style="color:rgba(255, 255, 255, 0.5);">(<i>n</i>‚Äâ=‚Äâ5)</span></div>
					<ul class="bulletList">
						<li>I‚ÄëŒ±<sub>RS</sub></li>
						<li>I‚ÄëŒ±<sub>LS</sub></li>
						<li>II‚ÄëŒ±<sub>RS</sub></li>
						<li>II‚ÄëŒ±<sub>LS</sub></li>
						<li>I‚ÄëŒ±<sub>LU</sub></li>
						<li>I‚ÄëŒ±<sub>RU</sub></li>
						<li>II‚ÄëŒ±<sub>LU</sub></li>
						<li>II‚ÄëŒ±<sub>RU</sub></li>
						<li>I‚ÄëŒ±<sub>C</sub></li>
					</ul>
				</div>
				<div id="betaturns" class="structureBox">
					<div style="text-align:center;pointer-events:none;">Œ≤-Turns<br><span style="color:rgba(255, 255, 255, 0.5);">(<i>n</i>‚Äâ=‚Äâ4)</span></div>
					<ul class="bulletList">
						<li>I</li>
						<li>I‚Äô</li>
						<li>II</li>
						<li>II‚Äô</li>
						<li>IV<sub>1&nbsp;to&nbsp;4</sub></li>
						<li>VI<sub>a1</sub></li>
						<li>VI<sub>a2</sub></li>
						<li>VI<sub>b</sub></li>
						<li>VIII</li>
					</ul>
				</div>
				<div id="gammaturns" class="structureBox">
					<div style="text-align:center;pointer-events:none;">Œ≥-Turns<br><span style="color:rgba(255, 255, 255, 0.5);">(<i>n</i>‚Äâ=‚Äâ3)</span></div>
					<ul class="bulletList">
						<li>Classic</li>
						<li>Inverse</li>
					</ul>
				</div>
				<div id="epsilonturns" class="structureBox">
					<div style="text-align:center;pointer-events:none;">…õ-Turns<br><span style="color:rgba(255, 255, 255, 0.5);">(<i>n</i>‚Äâ=‚Äâ3)</span></div>
					<ul class="bulletList">
						<li>Family&nbsp;1</li>
						<li>Family&nbsp;2</li>
						<li>Family&nbsp;3</li>
						<li>Family&nbsp;4</li>
						<li>Family&nbsp;5</li>
						<li>Family&nbsp;6</li>
					</ul>
				</div>
				<div id="deltaturns" class="structureBox">
					<div style="text-align:center;pointer-events:none;">Œ¥-Turns<br><span style="color:rgba(255, 255, 255, 0.5);">(<i>n</i>‚Äâ=‚Äâ2)</span></div>
				</div>
			</div>	
			<div id="coil" class="structureBox singleItem" "="">
				<div id="randomcoil_title" class="structureBoxTitle">Random Coil</div>
			</div>
	</div>

	<div id="secondarystructures_content" style="display:block;" class="sscontent">
		<div id="secondarystructures_interactiveContent" style="margin-right:20px;">
			<h2>Secondary Structure Elements</h2>
		</div>

		<div id="secondarystructures_explainingText" class="explainingtext ">
			<p>In brief, secondary structure refers to the local geometry of the polypeptide backbone and well-defined, often repetitive patterns of stabilising hydrogen bonds. Based on these criteria, different classes of secondary structure elements can be defined. To the left you can see an overview of most of these classes and subclasses described in the literature.</p>
			
			<p>There are many known subclasses, some of which (shown in <span style="color:#e9c155;"><b>blue</b></span>) are not even included in any of the tools available for secondary structure decomposition analysis for far-UV CD data. This is one of the reasons why results obtained with this traditional data analysis approach should be considered estimates rather than accurate, definite answers.</p>
			
			<p>In fact, not even secondary structure content based on 3d models obtained from techniques such as x-ray crystallography or nuclear magnetic resonance (NMR) spectroscopy is definite. Secondary structure assignments based on 3d structures are most often carried out using the DSSP algorithm developed by Kabsch and Sander in 1983. However, other valid algorithms for this purpose exist that make use of different criteria and won't yield the exact same assignments (<a href="https://secondary-structure.netlify.app/#Martin" onclick="reference(this.href);">Martin et al. 2005</a>). Secondary structure assignments based on such algorithms are always employed for the compilation of reference far-UV CD data sets that secondary structure decomposition analysis relies upon. This poses another factor that ultimately affects the accuracy of this kind of analysis.</p>
			 <!--- Write something about the color code / which sub structures can or cannot be identified based on CD data ---->
		</div>
	</div>

	<div id="helices_content" style="display:none;" class="sscontent">
		<div id="helices_interactiveContent" style="margin-right:20px;">
			<h2>Helices</h2>
			<a id="helices_backbone_toggle" data-backbone="hidden" onclick="setView(&#39;idealalpha&#39;, &#39;6&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Show backbone atoms</a>
			<a id="helices_sidechains_toggle" data-sidechains="visible" onclick="setView(&#39;idealalpha&#39;, &#39;1&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Hide sidechains</a>
			<a id="helices_hbonds_toggle" data-hbonds="hidden" onclick="setView(&#39;idealalpha&#39;, &#39;2&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Show H-bonds</a>
			<a id="helices_axis_toggle" data-axis="visible" onclick="setView(&#39;idealalpha&#39;, &#39;3&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Hide helical axis</a>
			<a id="helices_termini_toggle" data-termini="hidden" onclick="setView(&#39;idealalpha&#39;, &#39;4&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Show termini &amp; caps</a>
			<a id="helices_dipoles_toggle" data-dipoles="hidden" onclick="setView(&#39;idealalpha&#39;, &#39;5&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Show dipole moments</a>

			<div id="JmolDiv_helices" class="jmolContainer"></div>
		</div>
		<!--- XXX Maybe write something about the often amphiphilic nature of helices? Or maybe rather leave that for tertiary structure, because it makes more sense explaining this in context. --->
		<div id="helices_explainingText" class="explainingtext">
			<p id="helices_text1" data-active="false" class="hoverable">üñ∞ Helices in polypetides comprised of L-amino acids are virtually always right-handed, i.e., when looking down the <span style="color:#e9c155;"><b>helical axis</b></span> and tracing the trajectory of the <span style="color:#67cbdf;"><b>main chain</b></span> into the distance, the <span style="color:#212b38;"><b>projection</b></span> of this trace will follow a clockwise direction. The <span style="color:#7a73bf;"><b>side chains</b></span> of the amino acid residues are more or less staggered and point away from the helical axis. </p>
			<p id="helices_text2" data-active="false" class="hoverable">üñ∞ The main chain of a helix is stabilised by intrahelical <span style="color:#f79424;"><b>hydrogen bonds</b></span>. A helix consisting of <i>n</i> residues will always only have up to <i>n</i>&nbsp;-&nbsp;4 hydrogen bonds because neither the N-H groups of the four residues at the start (<span style="color:#005a9d;"><b>N-terminus</b></span>) nor the C=O groups of the four residues at the end (<span style="color:#b51726;"><b>C-terminus</b></span>) of the helix participate in intrahelical hydrogen bonds. The very first and very last residue of a helix is called <span style="color:#0000ff;"><b>N-Cap</b></span> and <span style="color:#ff0000;"><b>C-Cap</b></span>, respectively.</p>
			
			<p id="helices_text3" data-active="false" class="hoverable">üñ∞ Due to the relatively high electronegativity of <span style="color:#0000ff;"><b>nitrogen</b></span> and <span style="color:#ff0000;"><b>oxygen</b></span>, the N-H and C=O groups of peptide bonds have partial charges which result in a strong dipole moment. As all peptide groups point more or less into the same direction parallel to the helical axis, the <span style="color:#000000;"><b>amide dipole moments</b></span> are in near perfect alignment and cumulatively form a helix macrodipole where the N‚Äëterminus and C‚Äëterminus have unit charges of approximately Œ¥<sup>+</sup>&nbsp;=&nbsp;+0.5 and Œ¥<sup>-</sup>&nbsp;=&nbsp;-0.5, respectively.</p>
			
			<p>Three types of helices can be defined based on the number <i>n</i> of peptide bonds between any two residues that provide the donor (<i>i</i> + <i>n</i>) and acceptor group (<i>i</i>) for a hydrogen bond:
			</p><ul><li>œÄ-helices (<i>i</i> + 5 ‚Üí <i>i</i>)</li>
			<li>Œ±-Helices (<i>i</i> + 4 ‚Üí <i>i</i>)</li>
			<li>3<sub>10</sub>-helices (<i>i</i> + 3 ‚Üí <i>i</i>)
			</li></ul>
			On average, Œ±-helices account for about a third of the secondary structures observed in proteins, whereas 3<sub>10</sub>-helices and œÄ-helices are less abundant, contributing to about 4% and less than 1% of average protein secondary structure, respectively. Short segments of these rarer types of helices are often found at the ends of Œ±-helices.<p></p>
			
		</div>
	</div>

	<div id="pihelix_content" style="display:none;" class="sscontent">
		<div id="pihelix_interactiveContent" style="margin-right:20px;">
			<h2>œÄ-Helices</h2>
			<a onclick="setView(&#39;2QD1&#39;, &#39;1&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Reset View</a>
			<a id="pi_hbonds_toggle" data-hbonds="hidden" onclick="setView(&#39;2QD1&#39;, &#39;2&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Show backbone with H-bonds</a>

			<div id="JmolDiv_pi" class="jmolContainer"></div>
		</div>
		<div id="pihelix_explainingText" class="explainingtext">
			<p>In œÄ-helices, hydrogen bonds are formed between the amino group of one residue (<i>i</i>) and the carbonyl group of the residue five residues upstream in the sequence (<i>i</i>&nbsp;-&nbsp;5).</p><p>œÄ-Helices follow certain overall charasteristics:</p>
			<ul>
				<li>Any two <b>consecutive residues</b> are
					<ul>
						<li>translated by 1.15 √Ö relative to each other.</li>
						<li>rotated around the helical axis by 87¬∞ relative to each other.</li>
					</ul>
				</li>
				
				<li>A <b>single turn</b> in the helix 
					<ul>
						<li>corresponds to 4.4 amino acids.</li>
						<li>has a pitch (i.e., vertical distance between consecutive turns) of 1.15 √Ö ¬∑ 4.4 = 5.1 √Ö.</li>
					</ul>
				</li><li>The <b>dihedral angles</b> are typically close to œÜ ‚âà -57¬∞ and œà ‚âà -70¬∞.</li>
			</ul>
		</div>
	</div>

	<div id="regular_content" style="display:none;" class="sscontent">
		<div id="regular_interactiveContent" style="margin-right:20px;">
			<h3>Regular Œ±-Helices</h3>
			<a onclick="setView(&#39;6EC0&#39;, &#39;1&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Reset View</a>
			<a id="regular_hbonds_toggle" data-hbonds="hidden" onclick="setView(&#39;6EC0&#39;, &#39;2&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Show backbone with H-bonds</a>

			<div id="JmolDiv_regular" class="jmolContainer"></div>
		</div>
		<div id="regular_explainingText" class="explainingtext">
			<p>Hydrogen bonds of Œ±-Helices are formed between the amino group of one residue (<i>i</i>) and the carbonyl group of the residue four residues upstream in the sequence (<i>i</i>&nbsp;-&nbsp;4).</p><p>Regular Œ±-Helices are more or less straight, have an average length of 10 to 12 residues, and follow certain overall charasteristics:</p>
			<ul>
				<li>Any two <b>consecutive residues</b> are
					<ul>
						<li>translated by 1.5 √Ö relative to each other.</li>
						<li>rotated around the helical axis by 100¬∞ relative to each other.</li>
					</ul>
				</li>
				
				<li>A <b>single turn</b> in the helix 
					<ul>
						<li>corresponds to 3.6 amino acids.</li>
						<li>has a pitch (i.e., vertical distance between consecutive turns) of 1.5 √Ö ¬∑ 3.6 = 5.4 √Ö.</li>
					</ul>
				</li><li>The <b>dihedral angles</b> are typically close to œÜ ‚âà -58¬∞ and œà = -47¬∞.</li>
			</ul>
		</div>
	</div>

	<div id="distorted_content" style="display:none;" class="sscontent">
		<div id="distorted_interactiveContent" style="margin-right:20px;">
			<h3>Distorted Œ±-Helices</h3>
			<button class="button" onclick="distorted(-1)">‚Üê</button>
			<span id="PDB_ID_distorted"></span>
			<button class="button" onclick="distorted(1)">‚Üí</button>
			<br><span id="caption_distorted"></span>

			<div id="JmolDiv_distorted" class="jmolContainer"></div>
		</div>
		<div id="distorted_explainingText" class="explainingtext">
			<p>
			Canonical Œ±-helices hardly exist, as dihedral angles and other characteristics rarely conform to strict textbook definitions. Deviations from ideal conformation often observed include:
			</p><ul>
				<li>Curvature of the helical axis (~60&nbsp;√Ö radius, <a href="https://secondary-structure.netlify.app/#Barlow" onclick="reference(this.href);">Barlow et al. 1988</a>)</li>
				<li>Splits, breaks, kinks, or bends</li>
				<li>Distortions located at the helix 
				<ul>
				<li>caps, i.e., first (N-cap) or last (C-cap) residue of the helix</li>
				<li>termini, i.e., first (N-terminus) or last (C-terminus) few residues at either end of the helix</li>
				</ul>
				</li>
			</ul>
			Such irregularities can involve dihedral angles that are more typical for other secondary structure elements. On average, about one third of residues in Œ±-helices can be considered distorted.
			<p></p>
			<p>
			Atypical helix conformations are promoted by certain amino acids because their side chains impose restictions on the torsion angles of the main chain, as otherwise steric clashes would occur. This includes, for example, amino acids with side chains that are branched at the Œ≤-carbon atom such as threonine or valine. Due to their inferior capability to stabilise a helical structure, these amino acids have a lower propensity to occur within Œ±-helices.
			</p>
			<p>
			Breaks or kinks are often introduced by proline because its backbone nitrogen is unavailable for the formation of hydrogen bonds.
			</p>
		</div>
	</div>

	<div id="threeten_content" style="display:none;" class="sscontent">
		<div id="threeten_interactiveContent" style="margin-right:20px;">
			<h2>3<sub>10</sub>-Helices</h2>
			<a onclick="setView(&#39;1GPB&#39;, &#39;1&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Reset View</a>
			<a id="threeten_hbonds_toggle" data-hbonds="hidden" onclick="setView(&#39;1GPB&#39;, &#39;2&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Show backbone with H-bonds</a>

			<div id="JmolDiv_310" class="jmolContainer"></div>
		</div>
		<div id="threeten_explainingText" class="explainingtext">
			<p>In 3<sub>10</sub>-helices, hydrogen bonds are formed between the amino group of one residue (<i>i</i>) and the carbonyl group of the residue three residues upstream in the sequence (<i>i</i>&nbsp;-&nbsp;3).</p><p>3<sub>10</sub>-Helices are typically short with an average length of 3 to 5 residues and follow certain overall charasteristics:</p>
			<ul>
				<li>Any two <b>consecutive residues</b> are
					<ul>
						<li>translated by 2.0 √Ö relative to each other.</li>
						<li>rotated around the helical axis by 120¬∞ relative to each other.</li>
					</ul>
				</li>
				
				<li>A <b>single turn</b> in the helix 
					<ul>
						<li>corresponds to 3 amino acids.</li>
						<li>has a pitch (i.e., vertical distance between consecutive turns) of 2 √Ö ¬∑ 3 = 6 √Ö.</li>
					</ul>
				</li><li>The <b>dihedral angles</b> are typically close to œÜ ‚âà -49¬∞ and œà ‚âà -26¬∞.</li> <!-- -74 / -4?!-->
			</ul>
			<p>3<sub>10</sub>-Helices make up about 10 to 15% of all helices. Their name originates from a historical designation based on the number of residues per helical turn (3) and the number of backbone atoms (10) between a hydrogen bond's donor and acceptor groups.</p>
		</div>
	</div>

	<div id="strands_content" style="display:none;" class="sscontent">
		<div id="strands_interactiveContent" style="margin-right:20px;">
			<h2>Strands</h2>
			<a onclick="setView(&#39;2TRX&#39;, &#39;0&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Reset View</a>
			<a id="strands_sidechains_toggle" data-sidechains="visible" onclick="setView(&#39;2TRX&#39;, &#39;1&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Hide sidechains</a>
			<a id="strands_hbonds_toggle" data-hbonds="visible" onclick="setView(&#39;2TRX&#39;, &#39;2&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Hide H-bonds</a>

			<div id="JmolDiv_strands" class="jmolContainer"></div>
		</div>
		<div id="strands_explainingText" class="explainingtext">
			<p>In Œ≤-strands, the backbone has an extended conformation with dihedral angles of ideally œÜ&nbsp;‚âà&nbsp;-120¬∞ and œà&nbsp;‚âà&nbsp;+120¬∞, which might be conceptualised as a stretched-out helix with two residues per turn.</p>
			<p id="strands_text1" data-active="false" class="hoverable">üñ∞ Within a single Œ≤-strand, i.e., a segment of limited length with this conformation, the amide and carbonyl groups are too far apart to form hydrogen bonds with each other. Therefore, Œ≤-strands are virtually never found in isolation; instead, multiple strands align alongside each other so that <span style="color:#f79424;"><b>hydrogen bonds</b></span> can form between them and stabilise a larger structure, the Œ≤-sheet.</p>
			<p id="strands_text2" data-active="false" class="hoverable">üñ∞ Parallel and anti-parallel Œ≤-sheets can be distinguished, depending on which direction strands are aligned in relative to each other. However, mixed Œ≤-sheets that include both are possible, as shown here for human thioredoxin.</p>
			<p id="strands_text3" data-active="false" class="hoverable">üñ∞ Œ≤-Sheets are sometimes called "pleated" Œ≤-sheets because consecutive C<sub>Œ±</sub>-atoms and their corresponding <span style="color:#7a73bf;"><b>side chains</b></span> are alternatingly located on either side of the sheet's plane.</p>
			<p>On average, Œ≤-strands span 6 residues, and most Œ≤-sheets consist of fewer than 6 strands. About a quarter of the residues in known protein structures have a beta-sheet conformation.</p>
		</div>
	</div>

	<div id="parallel_content" style="display:none;" class="sscontent">
		<div id="parallel_interactiveContent" style="margin-right:20px;">
			<h2>Parallel Œ≤-Strands</h2>
			<a onclick="setView(&#39;2BEG&#39;, &#39;1&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Reset View</a>

			<div id="JmolDiv_parallel" class="jmolContainer"></div>
		</div>
		<div id="parallel_explainingText" class="explainingtext">
			<p>It is possible for more distant parts of an amino acid sequence to form strands and come together with the same direction to form parallel Œ≤-sheets. On average, the dihedral angles in untwisted parallel Œ≤-sheets are close to œÜ&nbsp;‚âà&nbsp;-116¬∞ and œà&nbsp;‚âà&nbsp;+118¬∞ (<a href="https://secondary-structure.netlify.app/#Nesloney" onclick="reference(this.href);">Nesloney et al. 1996</a>).</p>
			
			<p>The <span style="color:#e9c155;"><b>distances</b></span> between hydrogen bonds formed between adjacent strands are more or less equivalent in parallel Œ≤-sheets. Moreover, the <span style="color:#f79424;"><b>hydrogen bonds</b></span> in parallel Œ≤-sheets are neither perpendicular to the strands nor parallel to each other. Due to this hydrogen bonding geometry, parallel Œ≤-sheets are less stable than anti-parallel Œ≤-sheets, which might be the reason why parallel Œ≤-sheets usually contain at least four strands and don't show the same propensity for twists or bulges.</p>
		</div>
	</div>

	<div id="lefttwist_content" style="display:none;" class="sscontent">
		<div id="lefttwist_interactiveContent" style="margin-right:20px;">
			<h3>Left-Twisted Anti-Parallel Œ≤-Strands</h3>
			<a onclick="setView(&#39;1OFS&#39;, &#39;1&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Reset View</a>

			<div id="JmolDiv_lefttwist" class="jmolContainer"></div>
		</div>
		<div id="lefttwist_explainingText" class="explainingtext">
			<p>Left-twisted Œ≤-sheets are virtually never observed because right-twisted Œ≤-sheets are energetically favoured. This is, in part, due to steric hindrance that would occur in left-twisted Œ≤-sheets, but also due to hydrogen bonding (<a href="https://secondary-structure.netlify.app/#Shamovsky" onclick="reference(this.href);">Shamovsky et al. 2000</a>) and electrostatic effects (<a href="https://secondary-structure.netlify.app/#Maccallum" onclick="reference(this.href);">Maccallum et al. 1995</a>).</p>
			
			<p>The example shows Œ≤-strands in pea lectin that are slighty left-twisted at least locally. Left-handed twists usually involve glycine, because it is the only amino acid that lacks a Œ≤-carbon and, thus, cannot suppress a left-hand twist (<a href="https://secondary-structure.netlify.app/#Shamovsky" onclick="reference(this.href);">Shamovsky et al. 2000</a>).</p>
		</div>
	</div>

	<div id="relaxed_content" style="display:none;" class="sscontent">
		<div id="relaxed_interactiveContent" style="margin-right:20px;">
			<h3>Relaxed Anti-Parallel Œ≤-Strands</h3>
			<a onclick="setView(&#39;2YXF&#39;, &#39;1&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Reset View</a>

			<div id="JmolDiv_relaxed" class="jmolContainer"></div>
		</div>
		<div id="relaxed_explainingText" class="explainingtext">
			<p>A Œ≤-sheet often conists of strands that span the amino acid sequence sequentially and are separated only by short turns; this results in anti-parallel Œ≤-sheets, because the backbone changes direction as it folds onto itself.</p>
			<p>Relaxed, anti-parallel Œ≤-sheets form a pattern of alternating <span style="color:#e9c155;"><b>shorter</b></span> and <span style="color:#212b38;"><b>longer</b></span> distances between consecutive hydrogen bonds. Moreover, the <span style="color:#f79424;"><b>hydrogen bonds</b></span> are both perpendicular to the strands and more or less parallel to each other. On average, the dihedral angles in untwisted anti-parallel Œ≤-sheets are close to œÜ&nbsp;‚âà&nbsp;-145¬∞ and œà&nbsp;‚âà&nbsp;+140¬∞ (<a href="https://secondary-structure.netlify.app/#Nesloney" onclick="reference(this.href);">Nesloney et al. 1996</a>).</p>
			<p>Truly relaxed Œ≤-sheets are not that common because most Œ≤-sheets are at least slighty right-twisted.</p>
		</div>
	</div>

	<div id="righttwist_content" style="display:none;" class="sscontent">
		<div id="righttwist_interactiveContent" style="margin-right:20px;">
			<h3>Right-Twisted Anti-Parallel Œ≤-Strands</h3>
			<a onclick="setView(&#39;1BA7&#39;, &#39;1&#39;)" style="cursor: pointer; cursor: hand;color:#67cbdf;text-decoration:underline;">Reset View</a>

			<div id="JmolDiv_righttwist" class="jmolContainer"></div>
		</div>
		<div id="righttwist_explainingText" class="explainingtext">
			<p>Hardly any anti-parallel Œ≤-sheet is actually planar; most anti-parallel Œ≤-sheets have a twist which results from a relative right-handed rotation of 30¬∞ per residue.</p>
			
			<p>Definition of the twist sense (<a href="https://secondary-structure.netlify.app/#Weatherford" onclick="reference(this.href);">Weatherford et al. 1979</a>) can sometimes seem a bit confusing, so let's have a look at the example for a right-twisted Œ≤-sheet here.</p>

			<p id="righttwist_text1" data-active="false" class="hoverable">üñ∞ Firstly, the <span style="color:#e9c155;"><b>individual Œ≤-strands</b></span> are right-twisted, i.e., looking down along the polypeptide chain, a clockwise rotation around the axis parallel to the Œ≤-strand is apparent.</p>
			
			<p id="righttwist_text2" data-active="false" class="hoverable">üñ∞ Similarly, multiple Œ≤-strands make up a <span style="color:#212b38;"><b>right-twisted Œ≤-sheet</b></span>, i.e., looking down along an axis with the same orientation as the Œ≤-strands, a clockwise rotation of the <span style="color:#f79424;"><b>hydrogen bonds</b></span> around this axis is apparent.</p>
			
			<p id="righttwist_text3" data-active="false" class="hoverable">üñ∞ Confusingly, right-twisted Œ≤-sheets are sometimes ascribed a left twist; this occurs if a different reference axis is chosen that is orthogonal to the orientation of the Œ≤-strands: when looking along the direction of the hydrogen bonds, a <span style="color:#7a73bf;"><b>left-handed interchain twist</b></span>, i.e., across the Œ≤-strands, is apparent.</p>
			
			<p>On average, the dihedral angles in heavily twisted anti-parallel Œ≤-sheets are close to œÜ&nbsp;‚âà&nbsp;-96¬∞ and œà&nbsp;‚âà&nbsp;+158¬∞ (<a href="https://secondary-structure.netlify.app/#Nesloney" onclick="reference(this.href);">Nesloney et al. 1996</a>).</p>
		</div>
	</div>

	<div id="bulge_content" style="display:none;" class="sscontent">
		<div id="bulge_interactiveContent" style="margin-right:20px;">
			<h2>Œ≤-Bulges</h2>
			<button class="button" onclick="bulge(-1)">‚Üê</button>
			<span id="bulge_type"></span> <span id="PDB_ID_bulge"></span>
			<button class="button" onclick="bulge(1)">‚Üí</button>
			<br><span id="caption_bulge"></span>

			<div id="JmolDiv_bulge" class="jmolContainer"></div>
		</div>
		<div id="bulge_explainingText" class="explainingtext">
			<p>Œ≤-Bulges have been characterised well in the literature, so we shall quote a primary source here. As described in <a href="https://secondary-structure.netlify.app/#Richardson" onclick="reference(this.href);">Richardson et al. 1978</a>, "a Œ≤ bulge is a region between two consecutive Œ≤-type hydrogen bonds which includes two residues (positions 1 and 2) on one strand opposite a single residue (position x) on the other strand."</p>
			<p>"Compared to regular Œ≤ structure, a Œ≤ bulge puts the usual alternation of side-chain direction out of register on one of the strands, introduces a slight bend in the Œ≤ sheet, and locally accentuates the usual right-handed strand twist. Almost all Œ≤ bulges are between antiparallel strands, usually between a narrow rather than a wide pair of hydrogen bonds."</p>

			<p>And furthermore, "a Œ≤ bulge is defined as a region between two consecutive Œ≤-type hydrogen bonds which includes two residues on one strand opposite a single residue on the other strand. [...] In more than 80% of the cases, Œ≤ bulges occur between a closely spaced pair of hydrogen bonds rather than a widely spaced pair. Œ≤ bulges are extremely rare in parallel Œ≤ structure."</p>
			
			<p>"[...] an additional Œ≤ strand cannot continue to bond past the bulge on the two-residue side, so that a bulge is always at either an edge or an end of the Œ≤ sheet. The additional backbone length of the extra residue on the bulged strand is accommodated partly by bulging that strand up and out and partly by putting a slight bend in the Œ≤ sheet. The distance between Œ± carbons <i>n</i> and <i>n</i> + 3 on either end of the two-residue bulged side averages 8.0 √Ö."</p>
		</div>
	</div>

	<div id="turns_content" style="display:none;" class="sscontent">
		<div id="turns_interactiveContent" style="margin-right:20px;">
			<h2>Turns</h2>

			<div id="JmolDiv_turns" class="jmolContainer"></div>
		</div>
		<div id="turns_explainingText" class="explainingtext">
			<p>Turns are structural segments that reverse the overall directionality of the main chain; "Turns comprise <i>n</i> consecutive residues (denoted <i>i</i> to <i>i</i>&nbsp;+&nbsp;<i>n</i>), in which the distance between C<sub>Œ±</sub>(s) of residues <i>i</i> and <i>i</i>&nbsp;+&nbsp;<i>n</i> must be smaller than 7‚Äâ√Ö [...]. [...] The restrictive distance between C<sub>Œ±</sub>s applies a particular geometry to the backbone, thereby causing it to turn back on itself." (<a href="https://secondary-structure.netlify.app/#deBrevern" onclick="reference(this.href);">de Brevern et al. 2016</a>)</p>
			
			<p>Turns are typically stabilised by intramolecular backbone-backbone <span style="color:#f79424;"><b>hydrogen bonds</b></span>, and there are turns that are characterised by hydrogen bonds between the amine hydrogen (N-H) of one residue and the carbonyl oxygen (C=O) of a residue <i>upstream</i> in the amino acid sequence, as well as turns with hydrogen bonds between the N-H of a residue and the C=O of a residue <i>downstream</i> in the amino acid sequence.</p>
			
			<table style="border-spacing: 0px;">
			  <tbody><tr>
			    <th style="padding-right: 25px;">Turns with C=O¬∑¬∑¬∑H-N H-bonds:</th>
			    <th>Turns with N-H¬∑¬∑¬∑O=C H-bonds:</th>
			  </tr>
			  <tr>
			    <td style="vertical-align: top;">
				<ul>
				<li>œÄ-turns (<i>n</i>‚Äâ=‚Äâ6)</li>
				<li>Œ±-turns (<i>n</i>‚Äâ=‚Äâ5)</li>
				<li>Œ≤-turns (<i>n</i>‚Äâ=‚Äâ4)</li>
				<li>Œ≥-turns (<i>n</i>‚Äâ=‚Äâ3)</li>
				</ul>
			    </td>
			    <td style="vertical-align: top;">
				<ul>
				<li>…õ-turns (<i>n</i>‚Äâ=‚Äâ3)</li>
				<li>Œ¥-turns (<i>n</i>‚Äâ=‚Äâ2)</li>
				</ul>
			    </td>
			  </tr>
			</tbody></table>

			<p>For most types of turns, multiple classes and subclasses can be differentiated based on dihedral angles of the peptide bonds, i.e., backbone torsion angles of the amino acids involved.</p>
			
			<p>Turns that are longer, i.e., œÄ-turns and Œ±-turns in particular, may incorporate smaller internal turns. For example, there can be multiple overlapping Œ±- or Œ≤-turns embedded in a œÄ-turn, and Œ±-turns can include tighter Œ≤-turns or a Œ≥-turn (as is the case for Œ±-turns of the II‚ÄëŒ±<sub>LS</sub> subclass).</p>
			
			<p>An example for this is shown: this œÄ-turn comprises residues <span style="color:#67cbdf;"><b><i>i</i> to <i>i</i>&nbsp;+&nbsp;<i>5</i></b></span> and includes two overlapping type I Œ≤-turns that span residues <span style="color:#e9c155;"><b><i>i</i> to <i>i</i>&nbsp;+&nbsp;<i>3</i></b></span> and <span style="color:#7a73bf;"><b><i>i + 1</i> to <i>i</i>&nbsp;+&nbsp;<i>4</i></b></span>, respectively.</p>
		</div>
	</div>

	<div id="piturns_content" style="display:none;" class="sscontent">
		<div id="piturns_interactiveContent" style="margin-right:20px;">
			<h2>œÄ-Turns</h2>
			<button class="button" onclick="turn(-1, &#39;piturn&#39;)">‚Üê</button>
			<span id="piturn_type"></span> <span id="PDB_ID_piturn"></span>
			<button class="button" onclick="turn(1, &#39;piturn&#39;)">‚Üí</button>
			<br><span id="caption_piturn"></span>

			<div id="JmolDiv_piturn" class="jmolContainer"></div>
		</div>
		<div id="piturns_explainingText" class="explainingtext">
			<p>œÄ-Turns, which are often found at the C-terminus of Œ±-helices, have originally been classified based on the conformation of the amino acid residue in position <i>i</i>&nbsp;+&nbsp;4 (<a href="https://secondary-structure.netlify.app/#Rajashankar" onclick="reference(this.href);">Rajashankar et al. 1996</a>). However, an alternative classification has been proposed later that takes into account the dihedral angles of all residues that are in the middle of the turn (<a href="https://secondary-structure.netlify.app/#Dasgupta" onclick="reference(this.href);">Dasgupta et al. 2008</a>).</p>
			<p>This new classification distinguishes between non-hydrogen-bonded (NHB) and hydrogen-bonded (HB) œÄ-turns (specifically referring to hydrogen bonds between the residues in positions <i>i</i> and <i>i</i>&nbsp;+&nbsp;5) and further separates the latter into turns that are isolated and those that are part of Œ±-helix termini ('Schellman motif'); a further distinction is made for the latter based on whether a second hydrogen bond is present or not:</p>
			<ul>
			<li>Non-hydrogen-bonded œÄ-turns (œÄ-NHB)</li>
			<li>Hydrogen-bonded œÄ-turns (œÄ-HB)
				<ul>
				<li>Isolated</li>
				<li>Schellman (SCH) motif
					<ul>
					<li>SCH<sub>+Œ≤</sub></li>
					<li>SCH<sub>-Œ≤</sub></li>
					</ul>
				</li>
				</ul>
			</li>
			</ul>

		</div>
	</div>

	<div id="alphaturns_content" style="display:none;" class="sscontent">
		<div id="alphaturns_interactiveContent" style="margin-right:20px;">
			<h2>Œ±-Turns</h2>
			<button class="button" onclick="turn(-1, &#39;alphaturn&#39;)">‚Üê</button>
			<span id="alphaturn_type"></span> <span id="PDB_ID_alphaturn"></span>
			<button class="button" onclick="turn(1, &#39;alphaturn&#39;)">‚Üí</button>
			<br><span id="caption_alphaturn"></span>

			<div id="JmolDiv_alphaturn" class="jmolContainer"></div>
		</div>
		<div id="alphaturns_explainingText" class="explainingtext">
			<p>Œ±-Turns comprise five amino acid residues; they introduce a chain reversal and are stabilised by a hydrogen bond between the first and last residue of the turn just like other turns.</p>
			
			<p>The structural definition of type I‚ÄëŒ±<sub>RS</sub> Œ±-turns implies an ambiguity in the classification of secondary structure elements: the typical dihedral angles of these Œ±-turns are similar to those of the basic repetitive unit of the right-handed Œ±-helix. Similarly, Œ±-turns of opposite handedness can be observed (type I‚ÄëŒ±<sub>LS</sub>), with dihedral angles that correspond to the basic repetitive unit of left-handed Œ±-helices (<a href="https://secondary-structure.netlify.app/#Pavone" onclick="reference(this.href);">Pavone et al. 1996</a>).</p>
			
			<p>Œ±-Turns are relatively common; on average, two Œ±-turns can be found per protein.</p>
		</div>
	</div>

	<div id="betaturns_content" style="display:none;" class="sscontent">

		<div id="betaturns_interactiveContent" style="margin-right:20px;">
			<h2>Œ≤-Turns</h2>
			<button class="button" onclick="turn(-1, &#39;betaturn&#39;)">‚Üê</button>
			<span id="betaturn_type"></span> <span id="PDB_ID_betaturn"></span>
			<button class="button" onclick="turn(1, &#39;betaturn&#39;)">‚Üí</button>
			<br><span id="caption_betaturn"></span>

			<div id="JmolDiv_betaturn" class="jmolContainer"></div>
		</div>
		<div id="betaturns_explainingText" class="explainingtext">
			<p>From all types of turns, Œ≤-turns have received the most attention in the literature. Multiple types of Œ≤-turns have been described, including corresponding mirror images (denoted with an apostrophe, ').</p>
		
			<p>While looking through the types of Œ≤-turns, you might wonder why numbering suggests that some types are missing. This is because some types have been defined in the past, but later excluded due to various reasons:</p>
			<ul>
			<li>Type III (and III') has been excluded because it is basically a distortion of type I (I') and corresponds to the basic repetititve unit of 3<sub>10</sub>-helices.</li>
			<li>Type V (and V') is rare and its definition is inaccurate.</li>
			<li>Type VII, which is associated with a kink, is also rare and poorly defined.</li>
			</ul>
		</div>
	</div>

	<div id="gammaturns_content" style="display:none;" class="sscontent">
		<div id="gammaturns_interactiveContent" style="margin-right:20px;">
			<h2>Œ≥-Turns</h2>
			<button class="button" onclick="turn(-1, &#39;gammaturn&#39;)">‚Üê</button>
			<span id="gammaturn_type"></span> <span id="PDB_ID_gammaturn"></span>
			<button class="button" onclick="turn(1, &#39;gammaturn&#39;)">‚Üí</button>
			<br><span id="caption_gammaturn"></span>

			<div id="JmolDiv_gammaturn" class="jmolContainer"></div>
		</div>
		<div id="gammaturns_explainingText" class="explainingtext">
			<p>Œ≥-Turns are a tripeptide conformation of which only two types exist, the classic Œ≥-turn and the inverse Œ≥-turn (<a href="https://secondary-structure.netlify.app/#Milner-White" onclick="reference(this.href);">Milner-White et al. 1988</a>).</p>
			
			<p>Both types of Œ≥-turns are characterised by a seven-membered pseudo-cycle which is formed by the donor and acceptor atoms of an intramolecular hydrogen bond and the main chain atoms between them.</p>

			<p>Note that only 6 member atoms of this pseudo-cycle are visible in the 3d model because the hydrogen of the amide group is not shown.</p>
		</div>
	</div>

	<div id="epsilonturns_content" style="display:none;" class="sscontent">
		<div id="epsilonturns_interactiveContent" style="margin-right:20px;">
			<h2>…õ-Turns</h2>
			<button class="button" onclick="turn(-1, &#39;epsilonturn&#39;)">‚Üê</button>
			<span id="epsilonturn_type"></span> <span id="PDB_ID_epsilonturn"></span>
			<button class="button" onclick="turn(1, &#39;epsilonturn&#39;)">‚Üí</button>
			<br><span id="caption_epsilonturn"></span>

			<div id="JmolDiv_epsilonturn" class="jmolContainer"></div>
		</div>
		<div id="epsilonturns_explainingText" class="explainingtext">
			<p> While …õ-turns are formed by three residues just like Œ≥-turns, they are structurally different as the direction of the stabilising hydrogen bond between the first and last residue is reversed, i.e., the donor amide group is upstream rather than downstream of the carbonyl acceptor group.</p>
			
			<p>The hydrogen bond in …õ-turns hence forms a pseudo-cycle with eleven member atoms rather than seven. However, a Œ≥-turn may actually be included in an …õ-turn (<a href="https://secondary-structure.netlify.app/#Toniolo_b" onclick="reference(this.href);">Toniolo et al. 2017</a>).</p>
			
			<p>…õ-Turns have been classified in six different families and can be found in proteins in isolation or as part of hairpin motifs that consist of a pair of anti-parallel Œ≤-strands.</p>
			
			<p></p>
		</div>
	</div>

	<div id="deltaturns_content" style="display:none;" class="sscontent">
		<div id="deltaturns_interactiveContent" style="margin-right:20px;">
			<h2>Œ¥-Turns</h2>
			<button class="button" onclick="turn(-1, &#39;deltaturn&#39;)">‚Üê</button>
			<span id="deltaturn_type"></span> <span id="PDB_ID_deltaturn"></span>
			<button class="button" onclick="turn(1, &#39;deltaturn&#39;)">‚Üí</button>
			<br><span id="caption_deltaturn"></span>

			<div id="JmolDiv_deltaturn" class="jmolContainer"></div>
		</div>
		<div id="deltaturns_explainingText" class="explainingtext">
			<p>The shorter a turn gets the fewer options there are for main-chain reversal due to steric constraints.</p>

			<p>However, even a Œ¥-turn, a turn consisting of only two residues, is possible: for example, if the turn begins with a flexible residue such as a glycine (GLY) which is followed by a proline (PRO) in a semi-extended conformation, and the pair assumes the <i>cis</i>-conformation (i.e., the dihedral angle of the peptide bond between the two residues is close to zero, œâ&nbsp;‚âà&nbsp;0¬∞), then sterical hindrance is not too large (<a href="https://secondary-structure.netlify.app/#Toniolo_a" onclick="reference(this.href);">Toniolo et al. 2015</a>).</p>
			
			<p>Just like …õ-turns, Œ¥-turns can be stabilised by a hydrogen bond that has a direction opposite to hydrogen bonds found in longer turns.</p>
		</div>
	</div>
	
	<div id="coil_content" style="display:none;" class="sscontent">
		<div id="coil_interactiveContent" style="margin-right:20px;">
			<h2>Random Coil</h2>

			<div id="JmolDiv_coil" class="jmolContainer"></div>
		</div>
		<div id="coil_explainingText" class="explainingtext">
			<p>'Random coil' usually refers to any local polypeptide structure that lacks a well-defined or periodic pattern of hydrogen bonds and thus cannot be ascribed to any of the other classes of secondary structure. This typically includes loops that connect segments with defined secondary structure. However, the term is somewhat misleading because segments with random coil structure usually are neither coiled nor randomly structured.</p>
			
			<p>It is true that random coil segments are often more flexible and their dihedral angles don't follow canonic definitions but rather populate a distribution of values that give rise to an ensemble of conformations (<a href="https://secondary-structure.netlify.app/#Smith" onclick="reference(this.href);">Smith et al. 1996</a>). This holds usually true for <span style="color:#e9c155;"><b>terminal segments</b></span> at the ends of the polypeptide chain and <span style="color:#212b38;"><b>solvent-exposed loops</b></span> at the protein surface.</p>
			
			<p>However, even random coil structure can be subjected to a certain level of classification and loops are not always necessarily highly flexible. For example, the <span style="color:#67cbdf;"><b>omega loop</b></span> is a non-regular structure comprising six or more residues that lacks well-defined dihedral angles or a pattern of hydrogen bonding, but is often found to have limited flexibility. It can be identified by the fact that the first and last residue of the loop are in close vicinity to each other; they act as hinge points and gives rise to the loop's eponymous shape that resembles a Greek omega, Œ© (<a href="https://secondary-structure.netlify.app/#Papaleo" onclick="reference(this.href);">Papaleo et al. 2016</a>).</p>
			
		</div>
	</div>
</div>

<h1>Defining Secondary Structure</h1>

<p>When we talk about the secondary structure of a polypeptide, we basically refer to the local three-dimensional arrangement of its amino acid main chain, or backbone, into a series of linear segments that are stabilised by hydrogen bonds in well-defined, repeating patterns.</p>

<p>Firstly, secondary structure can be defined based on gemometric criteria for the hydrogen bonds which are formed between amide (N-H) and carbonyl (C=O) groups in those segments: Typically, these criteria are as follows (<a href="https://secondary-structure.netlify.app/#Taylor" onclick="reference(this.href);">Taylor et al. 1984</a>):
</p><ul>
<li>A H¬∑¬∑¬∑O distance &lt; 2.5&nbsp;√Ö</li>
<li>A N¬∑¬∑¬∑O distance &lt; 3.5&nbsp;√Ö</li>
<li>A N-H¬∑¬∑¬∑O angle &gt; 110¬∞</li>
</ul>
<p></p>

<p>Secondly, secondary structure can be defined based on the more or less specific relative positions of backbone atoms. These can be described by means of torsion angles, i.e., dihedral angles in a stereochemical context, where both terms are usually being used synonymously.</p>

<h2>Torsion Angles</h2>

<p>A dihedral angle, in general, is the angle between two intersecting planes or half-planes, and a torsion angle, in particular, is the dihedral angle between two planes on either side of a chemical bond that each span across a set of three atoms, where both planes include the two atoms that form the bond as well as the atom before or after this pair of atoms. Upon rotating both ends on either side of a bond in a molecule against each other, the torsion angle for that bond changes.</p>

<p>In proteins and peptides, three types of torsion angles can be distinguished for:
</p><ul>
<li><b>The bond preceding a C<sub>Œ±</sub> atom, œÜ (phi):</b> C<sup><i>i</i>&nbsp;-&nbsp;1</sup>‚ÄîN<sup><i>i</i></sup>‚ÄîC<sub>Œ±</sub><sup><i>i</i></sup>‚ÄîC<sup><i>i</i></sup></li>
<li><b>The bond following a C<sub>Œ±</sub> atom, œà (psi):</b> N<sup><i>i</i></sup>‚ÄîC<sub>Œ±</sub><sup><i>i</i></sup>‚ÄîC<sup><i>i</i></sup>‚ÄîN<sup><i>i</i>&nbsp;+&nbsp;1</sup></li>
<li><b>The peptide bond, œâ (omega):</b> C<sub>Œ±</sub><sup><i>i</i></sup>‚ÄîC<sup><i>i</i></sup>‚ÄîN<sup><i>i</i>&nbsp;+&nbsp;1</sup>‚ÄîC<sub>Œ±</sub><sup><i>i</i>&nbsp;+&nbsp;1</sup></li>
</ul>
where <i>i</i> depicts the number of a residue (and <i>i</i>&nbsp;+&nbsp;1 is the subsequent residue in the amino acid sequence), the two planes span the first three atoms and the last three atoms, respectively, and rotation around the bond in the middle affects the corresponding torsion angle.

<p></p>

<p>The peptide bonds in polypeptides have a partial double-bond character due to the delocalisation of the lone pair of electrons on the nitrogen atom. Therefore, peptide bonds almost exclusively assume a <i>trans</i> configuration, and the corresponding dihedral angle, œâ, is virtually always close to 180¬∞.</p>

<p>As opposed to the torsion angle of the peptide bond, œÜ and œà have more degrees of freedom. However, <a href="https://secondary-structure.netlify.app/#Ramachandran" onclick="reference(this.href);">G. N. Ramachandran</a> showed in 1963 that not just any theoretically possible combination of œÜ and œà is found in proteins, but rather dihedral angles are subject to certain constraints (which were later refined, e.g., see <a href="https://secondary-structure.netlify.app/#Lovell" onclick="reference(this.href);">Lovell et al. 2003</a>) and can be classified accordingly:
</p><ul>
<li><b>Allowed combinations</b> that won't result in steric clashes between atoms</li>
<li><b>Favoured combinations</b> that allow for the formation of stabilising backbone hydrogen bonds</li>
</ul>

These constraints are the reason why secondary structure can be classified into certain reoccuring elements that are characterised by sets of specific (within margins) dihedral angles and give rise to specific patterns of hydrogen bonding.<p></p>

<h2>Interactive Widget</h2>

<p>Now, in case the above isn't very tangible to you, we give you a little playground below for a better understanding of dihedral angles and how they affect secondary structure. For information about what you can do here, please refer to the short how-to next to it.</p>

<!-- <p>Note that, although peptide bonds virtually always have an angle œâ of close to 180¬∞, you are allowed to change it in this model.</p> -->

<div class="mainBox">
<div class="loading" style="visibility: hidden;">
	<img id="loading-image" src="./cursor_wait.gif" alt="Loading...">
</div>
<div id="playground" class="img-noscale">
	<div id="JmolDiv_intro" class="playgroundContent img-noscale"><div id="jmol_intro_appletinfotablediv" style="width:100%;height:100%;position:relative;font-size:14px;text-align:left"> <img id="jmol_intro_waitimage" src="./cursor_wait.gif" style="position: absolute; bottom: 10px; left: 10px; z-index: 9006; display: none;"><div id="jmol_intro_appletdiv" style="z-index: 9001; width: 100%; height: 100%; position: absolute; top: 0px; left: 0px; background-repeat: no-repeat; background-position: center center;"><img width="0" height="0" src="./saved_resource" onerror="jmol_intro._cover(false)"><canvas tabindex="1" width="476" height="381" id="jmol_intro_canvas2d" style="width: 100%; height: 100%; z-index: 9002; cursor: default;"></canvas></div><div id="jmol_intro_2dappletdiv" style="position:absolute;width:100%;height:100%;overflow:hidden;display:none"></div><div id="jmol_intro_infotablediv" style="width: 100%; height: 100%; position: absolute; top: 0px; left: 0px; display: none;"><div id="jmol_intro_infoheaderdiv" style="height: 20px; width: 100%; background: yellow; display: none;"><span id="jmol_intro_infoheaderspan">Jmol._Canvas2D (Jmol) "jmol_intro"</span><span id="jmol_intro_infocheckboxspan" style="position:absolute;text-align:right;right:1px;"><a href="javascript:Jmol.showInfo(jmol_intro,false)">[x]</a></span></div><div id="jmol_intro_infodiv" style="position:absolute;top:20px;bottom:0px;width:100%;height:100%;overflow:auto"><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/core/package.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/core/corejmol.z.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/J/translation/PO.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/core/corescript.z.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">JSmol exec jmol_intro start applet null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol JavaScript applet jmol_intro__8766593424054585__ initializing</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue debug null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue logLevel null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue allowjavascript null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">AppletRegistry.checkIn(jmol_intro__8766593424054585__)</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">vwrOptions: </span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">{ "loadStructCallback":"moleculeLoaded","measureCallback":"measurementDone","messageCallback":"messageReceived","appletReadyCallback":"Jmol._readyCallback","applet":true,"name":"jmol_intro","syncId":"8766593424054585","bgcolor":"#FFFFFF","signedApplet":"true","platform":"J.awtjs2d.Platform","display":"jmol_intro_canvas2d","documentBase":"https://secondary-structure.netlify.app/","codePath":"https://secondary-structure.netlify.app/jsmol/jsmol/j2s/","fullName":"jmol_intro__8766593424054585__","statusListener":"[J.appletjs.Jmol object]" }</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">setting document base to "https://secondary-structure.netlify.app/"</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager opening url https://secondary-structure.netlify.app/jsmol/jsmol/j2s/J/translation/JmolApplet/de.po</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">548 translations loaded</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">(C) 2015 Jmol Development</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol Version: 16.3.3  2024-10-09 15:43</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">java.vendor: Java2Script (HTML5)</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">java.version: 2022-06-24 05:54:49 (JSmol/j2s)</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">os.name: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/144.0.0.0 Safari/537.36</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Access: ALL</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">memory: 0.0/0.0</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">processors available: 1</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">useCommandThread: false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">appletId:jmol_intro (signed)</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue emulate null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">defaults = "Jmol"</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue boxbgcolor null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue bgcolor #FFFFFF</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">backgroundColor = "#FFFFFF"</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue ANIMFRAMECallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue APPLETREADYCallback Jmol._readyCallback</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">APPLETREADYCallback = "Jmol._readyCallback"</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">StatusManager APPLETREADYcallback set f=Jmol._readyCallback</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue ATOMMOVEDCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue AUDIOCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue CLICKCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue DRAGDROPCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue ECHOCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue ERRORCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue EVALCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue HOVERCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue IMAGECallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue LOADSTRUCTCallback moleculeLoaded</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">LOADSTRUCTCallback = "moleculeLoaded"</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">StatusManager LOADSTRUCTcallback set f=moleculeLoaded</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue MEASURECallback measurementDone</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">MEASURECallback = "measurementDone"</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">StatusManager MEASUREcallback set f=measurementDone</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue MESSAGECallback messageReceived</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">MESSAGECallback = "messageReceived"</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">StatusManager MESSAGEcallback set f=messageReceived</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue MINIMIZATIONCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue MODELKITCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue PICKCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue RESIZECallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue SCRIPTCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue SELECTCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue SERVICECallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue STRUCTUREMODIFIEDCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue SYNCCallback null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue doTranslate null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">language=de</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue doTranslate null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Note -- Presence of message callback disables disable translation; to enable message translation use jmolSetTranslation(true) prior to jmolApplet()</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Note -- language translation disabled</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue popupMenu null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue script null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue loadInline null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol getValue load null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol applet jmol_intro__8766593424054585__ ready</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/core/corebio.z.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">script 1 started</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">disablePopupMenu = false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">perspectiveDepth = false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/J/shape/Halos.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">showHiddenSelectionHalos = false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager opening url https://secondary-structure.netlify.app/preLoadingScript.txt</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/core/coretext.z.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Loading...</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">cache cleared</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">callback LOADSTRUCT failed TypeError: Cannot read properties of null (reading 'length')</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/J/render/HalosRenderer.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/core/corescriptcmd.z.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/JM/BioExt.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/core/corescriptmath.z.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks Pdb</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 6 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Jmol</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">6 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: autobonding; use  autobond=false  to not generate bonds automatically</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 5 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/JV/DataManager.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/J/api/JmolDataManager.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks Pdb</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 6 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Jmol</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">6 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: autobonding; use  autobond=false  to not generate bonds automatically</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 2 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks Pdb</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 6 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Jmol</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">6 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: autobonding; use  autobond=false  to not generate bonds automatically</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 1 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks Pdb</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 6 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Jmol</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">6 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: autobonding; use  autobond=false  to not generate bonds automatically</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 2 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks Pdb</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 6 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Jmol</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">6 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: autobonding; use  autobond=false  to not generate bonds automatically</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 1 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks Pdb</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 6 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Jmol</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">6 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: autobonding; use  autobond=false  to not generate bonds automatically</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 1 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks Pdb</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 6 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Jmol</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">6 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: autobonding; use  autobond=false  to not generate bonds automatically</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 1 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks Pdb</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 6 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Jmol</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">6 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: autobonding; use  autobond=false  to not generate bonds automatically</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 1 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks Pdb</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 6 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Jmol</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">6 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: autobonding; use  autobond=false  to not generate bonds automatically</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 1 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks Pdb</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 6 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Jmol</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">6 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: autobonding; use  autobond=false  to not generate bonds automatically</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 1 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/core/coreprop.z.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/J/adapter/writers/PDBWriter.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/J/api/JmolWriter.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/JV/OutputManagerJS.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/JV/OutputManager.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">callback LOADSTRUCT failed TypeError: Cannot read properties of null (reading 'length')</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks Pdb</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 46 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Jmol</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">46 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: autobonding; use  autobond=false  to not generate bonds automatically</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol 16.3.3  2024-10-09 15:43 DSSP analysis for model 1.1 - null</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">&nbsp;&nbsp;&nbsp;&nbsp;</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">W. Kabsch and C. Sander, Biopolymers, vol 22, 1983, pp 2577-2637</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">&nbsp;&nbsp;&nbsp;&nbsp;</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">We thank Wolfgang Kabsch and Chris Sander for writing the DSSP software,</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">and we thank the CMBI for maintaining it to the extent that it was easy to</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">re-engineer in Java for our purposes. </span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">&nbsp;&nbsp;&nbsp;&nbsp;</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Second generation DSSP 2.0 is used in this analysis. See Int. J. Mol. Sci. 2014, 15, 7841-7864; doi:10.3390/ijms15057841.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">&nbsp;&nbsp;&nbsp;&nbsp;</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">All bioshapes have been deleted and must be regenerated.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">&nbsp;&nbsp;&nbsp;&nbsp;</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 4 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">&nbsp;&nbsp;&nbsp;&nbsp;</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager.getAtomSetCollectionFromFile(https://files.rcsb.org/ligands/download/CYS.cif)</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager opening url https://files.rcsb.org/ligands/download/CYS.cif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks MMCif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/J/adapter/readers/cif/MMCifReader.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/J/adapter/readers/cif/CifReader.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/JU/CifDataParser.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">CYSTEINE</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">C3 H7 N O2 S</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">CYS</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for openFile(https://files.rcsb.org/ligands/download/CYS.cif): 1475 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 14 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Babel</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">14 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: not autobonding; use  forceAutobond=true  to force automatic bond creation</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 1 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/core/coresmiles.z.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/JS/SmilesStereo.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/JS/PolyhedronStereoSorter.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/JU/Eigen.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">loadScript jsmol/jsmol/j2s/javajs/api/EigenInterface.js</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.5482787027516816</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">mutating ALA1 to CYS</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">COMPARE using SMARTS [*.N][*.CA][*.C]</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.03274423262065588</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">RMSD = 0.03274423262065588 Angstroms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">&nbsp;&nbsp;&nbsp;&nbsp;</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager.getAtomSetCollectionFromFile(https://files.rcsb.org/ligands/download/HIS.cif)</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager opening url https://files.rcsb.org/ligands/download/HIS.cif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks MMCif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">HISTIDINE</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">C6 H10 N3 O2</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">HIS</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for openFile(https://files.rcsb.org/ligands/download/HIS.cif): 522 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 21 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Babel</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">21 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: not autobonding; use  forceAutobond=true  to force automatic bond creation</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 2 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.5293296027879072</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">mutating ALA2 to HIS</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">COMPARE using SMARTS [*.N][*.CA][*.C]</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.03491878015732151</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">RMSD = 0.03491878015732151 Angstroms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">&nbsp;&nbsp;&nbsp;&nbsp;</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager.getAtomSetCollectionFromFile(https://files.rcsb.org/ligands/download/ILE.cif)</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager opening url https://files.rcsb.org/ligands/download/ILE.cif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks MMCif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ISOLEUCINE</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">C6 H13 N O2</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ILE</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for openFile(https://files.rcsb.org/ligands/download/ILE.cif): 309 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 22 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Babel</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">22 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: not autobonding; use  forceAutobond=true  to force automatic bond creation</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 1 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.5481272150909605</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">mutating ALA3 to ILE</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">COMPARE using SMARTS [*.N][*.CA][*.C]</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.032492679679395446</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">RMSD = 0.032492679679395446 Angstroms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">&nbsp;&nbsp;&nbsp;&nbsp;</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager.getAtomSetCollectionFromFile(https://files.rcsb.org/ligands/download/ARG.cif)</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager opening url https://files.rcsb.org/ligands/download/ARG.cif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks MMCif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ARGININE</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">C6 H15 N4 O2</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ARG</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for openFile(https://files.rcsb.org/ligands/download/ARG.cif): 318 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 27 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Babel</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">27 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: not autobonding; use  forceAutobond=true  to force automatic bond creation</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 1 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.4887122775857622</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">mutating ALA4 to ARG</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">COMPARE using SMARTS [*.N][*.CA][*.C]</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.033213092972138765</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">RMSD = 0.033213092972138765 Angstroms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">&nbsp;&nbsp;&nbsp;&nbsp;</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager.getAtomSetCollectionFromFile(https://files.rcsb.org/ligands/download/SER.cif)</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager opening url https://files.rcsb.org/ligands/download/SER.cif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks MMCif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">SERINE</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">C3 H7 N O3</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">SER</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for openFile(https://files.rcsb.org/ligands/download/SER.cif): 343 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 14 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Babel</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">14 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: not autobonding; use  forceAutobond=true  to force automatic bond creation</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 1 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.5476991500716555</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">mutating ALA6 to SER</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">COMPARE using SMARTS [*.N][*.CA][*.C]</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.033094725101261845</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">RMSD = 0.033094725101261845 Angstroms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">&nbsp;&nbsp;&nbsp;&nbsp;</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager.getAtomSetCollectionFromFile(https://files.rcsb.org/ligands/download/CYS.cif)</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager opening url https://files.rcsb.org/ligands/download/CYS.cif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks MMCif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">CYSTEINE</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">C3 H7 N O2 S</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">CYS</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for openFile(https://files.rcsb.org/ligands/download/CYS.cif): 13 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 14 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Babel</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">14 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: not autobonding; use  forceAutobond=true  to force automatic bond creation</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 1 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.5481969085801209</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">mutating ALA7 to CYS</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">COMPARE using SMARTS [*.N][*.CA][*.C]</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.03277918436658359</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">RMSD = 0.03277918436658359 Angstroms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">&nbsp;&nbsp;&nbsp;&nbsp;</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager.getAtomSetCollectionFromFile(https://files.rcsb.org/ligands/download/ASN.cif)</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager opening url https://files.rcsb.org/ligands/download/ASN.cif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">The Resolver thinks MMCif</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ASPARAGINE</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">C4 H8 N2 O3</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ASN</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for openFile(https://files.rcsb.org/ligands/download/ASN.cif): 333 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">reading 17 atoms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: haveSymmetry:false haveUnitcells:false haveFractionalCoord:false</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 model in this collection. Use getProperty "modelInfo" or getProperty "auxiliaryInfo" to inspect them.</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Default Van der Waals type for model set to Babel</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">17 atoms created</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">ModelSet: not autobonding; use  forceAutobond=true  to force automatic bond creation</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Time for creating model: 2 ms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.5362164029523455</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">mutating ALA9 to ASN</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">COMPARE using SMARTS [*.N][*.CA][*.C]</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 mappings found</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">getSmilesCorrelation stddev=0.03301558023412085</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">RMSD = 0.03301558023412085 Angstroms</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap; color: red;">vwr handling error condition: TypeError: Cannot read properties of null (reading 'is')  </span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">script ERROR: TypeError: Cannot read properties of null (reading 'is') </span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">TypeError: Cannot read properties of null (reading 'is')</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">1 new bonds; 0 modified</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">37 atoms selected</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">hbondsRasmol = true</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">RasMol pseudo-hbond calculation</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">5 hydrogen bonds</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">66 atoms selected</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">FileManager opening url https://secondary-structure.netlify.app/postLoadingScript.txt</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">66 atoms selected</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Script completed</span></div><div style="white-space: nowrap;"><span style="white-space: nowrap;">Jmol script terminated</span></div></div></div></div></div>

	<div id="playgroundcontrols" class="playgroundContent img-noscale">
		<label for="sequence">Sequence:</label>
		<input onpaste="return false;" type="text" id="sequence" name="sequence" required="" maxlength="10" size="10" value="chirascan">

		<div>Default œÜ:<input type="range" class="slider" min="-180" max="180" value="-45" id="newPhiSlider"><span id="newPhiVal">-45</span>¬∞ <button type="button" id="phiButton" onclick="buttonClick(this.id)" disabled="">Apply to Selection</button></div>
		<div>Default œà:<input type="range" class="slider" min="-180" max="180" value="-60" id="newPsiSlider"><span id="newPsiVal">-60</span>¬∞ <button type="button" id="psiButton" onclick="buttonClick(this.id)" disabled="">Apply to Selection</button></div>
		<div style="display:none;"> <!-- For now, hide slider for omega -->
		<div>Default œâ:<input type="range" class="slider" min="-180" max="180" value="180" id="newOmegaSlider"><span id="newOmegaVal">180</span>¬∞ <button type="button" id="omegaButton" onclick="buttonClick(this.id)" disabled="">Apply to Selection</button></div>
		</div>

		<div id="selectedAngles" style="display:none;">
			<div><label for="selectedPhi">Selected œÜ:</label><input type="range" class="slider" min="-180" max="180" value="180" id="selectedPhiSlider"><span id="selectedPhiVal"></span>¬∞</div>
			<div><label for="selectedPsi">Selected œà:</label><input type="range" class="slider" min="-180" max="180" value="180" id="selectedPsiSlider"><span id="selectedPsiVal"></span>¬∞</div>
		</div>
		<div id="selectedOmegaAngle" style="display:none;"><label for="selectedOmega">Selected œâ:</label><input type="range" class="slider" min="-180" max="180" value="180" id="selectedOmegaSlider"><span id="selectedOmegaVal"></span>¬∞</div>

		<div>
			Planes:
			<input type="radio" id="noPlanes" name="anglePlanes" value="hide" onchange="anglePlanesChange()" checked="checked">None
			<input type="radio" id="phiPlanes" name="anglePlanes" value="phi" onchange="anglePlanesChange()">œÜ
			<input type="radio" id="psiPlanes" name="anglePlanes" value="psi" onchange="anglePlanesChange()">œà
			<input type="radio" id="omegaPlanes" name="anglePlanes" value="omega" onchange="anglePlanesChange()">œâ
		</div>

		<div>
			Bonds:
			<input type="radio" id="noBonds" name="angleBonds" value="hide" onchange="angleBondsChange()" checked="checked">None
			<input type="radio" id="phiBonds" name="angleBonds" value="phi" onchange="angleBondsChange()">œÜ
			<input type="radio" id="psiBonds" name="angleBonds" value="psi" onchange="angleBondsChange()">œà
			<input type="radio" id="omegaBonds" name="angleBonds" value="omega" onchange="angleBondsChange()">œâ
		</div>

		<div>
			Style:
			<input type="radio" id="ballstick" name="style" value="ballstick" onchange="styleChange()" checked="checked">Balls &amp; Sticks
			<input type="radio" id="cartoon" name="style" value="cartoon" onchange="styleChange()">Cartoon
		</div>

		<div>
			Clashes:
			<input type="radio" id="clashesFalse" name="clashes" value="false" onchange="clashesChange()" checked="checked">Hide
			<input type="radio" id="clashesTrue" name="clashes" value="true" onchange="clashesChange()">Show
		</div>
		
		<div>
			Extra labels in Ramachandran plot:
			<input type="radio" id="labelsFalse" name="labels" value="false" onchange="labelsChange()" checked="checked">Hide
			<input type="radio" id="labelsTrue" name="labels" value="true" onchange="labelsChange()">Show
		</div>
	</div>
	
	<div id="ramachandran_div" class="playgroundContent img-noscale">
		<canvas id="bgcanvas" width="381" height="381"></canvas>
		<canvas id="labelscanvas" width="381" height="381"></canvas>
		<canvas id="canvas" width="381" height="381"></canvas>
	</div>

	<div id="collapsibles" class="playgroundContent img-noscale">
		<h3>Usage How-To</h3>
		<button type="button" class="collapsible">View<span style="float:right;">+</span></button>
		<div class="content">
			<ul>
			<li><b>Pan:</b> Press Ctrl, click the right mouse button and drag.</li>
			<li><b>Rotate:</b> Click the left mouse button and drag.</li>
			<li><b>Zoom:</b> Click the middle mouse button and drag vertically, or use mouse wheel.</li>
			</ul>
		</div>
		<button type="button" class="collapsible">Peptide Sequence<span style="float:right;">+</span></button>
		<div class="content">
			<ul>
			<li><b>Add residues:</b> Type in one of the 20 single-letter amino acid codes.</li>
			<li><b>Remove single residues:</b> Use backspace or delete key to remove an individual amino acid.</li>
			<li><b>Remove multiple residues:</b> Select a part of the sequence, then hit backspace or delete key.</li>
			<li><b>Move residues:</b> Select a part of the sequence, then drag &amp; drop to the desired location.</li>
			<li><b>Replace residues:</b> Select a part of the sequence, then enter the single-letter code of an amino acid to replace with.</li>
			</ul>
		</div>
		<button type="button" class="collapsible">Dihedral Angles<span style="float:right;">+</span></button>
		<div class="content">
			<ul>
			<li><b>Default angles:</b> Use the Default sliders to change which dihedral angles any newly added amino acid will have.</li>
			<li><b>Change œÜ:</b> Select a single residue in the sequence, then use the Selected œÜ slider to change the residue's phi angle.</li>
			<li><b>Change œà:</b> Select a single residue in the sequence, then use the Selected œà slider to change the residue's psi angle.</li>
			<div style="display:none;"> <!-- For now, hide info for slider for omega -->
			<li><b>Change œâ:</b> Select two residues in the sequence, then use the Selected œâ slider to change œâ for the corresponding peptide bond.</li>
			</div>
			<li><b>Change angles for multiple amino acids:</b> Select a part of the sequence, then click one of the Apply to Selection buttons to change the dihedral angles of all the selected amino acids to the default.</li>
			<li><b>Fine tuning:</b> You can use the arrow keys to adjust sliders more precisely.</li>
			</ul>
		</div>
		<button type="button" class="collapsible">Display<span style="float:right;">+</span></button>
		<div class="content">
			<ul>
			<li><b>Planes:</b> Chose one of the options to display the planes for one of the three types of dihedral angles.</li>
			<li><b>Bonds:</b> Chose one of the options to highlight the bonds for one of the three types of dihedral angles.</li>
			<li><b>Style:</b> Chose one of the options to either show atoms in detail or a cartoon representation that depicts secondary structure.</li>
			<li><b>Clashes:</b> Visualise steric clashes; for clarity, only clashes between atoms of consecutive amino acids are considered.</li>
			</ul>
		</div>
		<button type="button" class="collapsible">Ramachandran Plot<span style="float:right;">+</span></button>
		<div class="content">
			<ul>
			<li><b>Drag:</b> Drag any point in the Ramachandran plot to change corresponding dihedral angles (this may be a bit slow as the 3d model is being recalculated accordingly). When dragging a point in the plot, the 3d model is automatically realigned.</li>
			<li><b>Similar angles:</b> There is one point for each residue in the plot; if multiple points have similar values, they are grouped together. Points corresponding to individual residues are labeled with the corresponding one-letter amino acid code, whereas grouped residues are labelled with a number indicating how many residues are in the group. To separate a residue from the group, select the residue in the sequence and change one of its dihedral angles.</li>
			<li><b>Extra labels:</b> Show additional labels in the Ramamchandran plot that indicate typical dihedral angles for a few selected secondary structure elements.</li>
			</ul>
		</div>
	</div>
</div>
</div>

<h1>References</h1>

<ol>
<li><a href="https://doi.org/10.1016/0022-2836(88)90641-9" target="_blank"><span id="Barlow">Barlow, D. J., &amp; Thornton, J. M. (1988). Helix geometry in proteins. Journal of Molecular Biology, 201(3), 601‚Äì619.</span></a></li>
<li><a href="https://doi.org/10.1038/srep33191" target="_blank"><span id="deBrevern">de Brevern, A. G. (2016). Extension of the classical classification of Œ≤-turns. Scientific Reports, 6(1).</span></a></li>
<li><a href="https://doi.org/10.1186/1472-6807-8-39" target="_blank"><span id="Dasgupta">Dasgupta, B., &amp; Chakrabarti, P. (2008). pi-Turns: types, systematics and the context of their occurrence in protein structures. BMC Structural Biology, 8(1), 39.</span></a></li>
<li><a href="https://doi.org/10.1002/prot.10286" target="_blank"><span id="Lovell">Lovell, S. C., Davis, I. W., Arendall, W. B., III, de Bakker, P. I. W., Word, J. M., Prisant, M. G., Richardson, J. S., &amp; Richardson, D. C. (2003). Structure validation by CŒ± geometry: œï,œà and CŒ≤ deviation. Proteins, 50(3), 437‚Äì450.</span></a></li>
<li><a href="https://doi.org/10.1016/s0022-2836(95)80057-3" target="_blank"><span id="Maccallum">Maccallum, P. H., Poet, R., &amp; James Milner-White, E. (1995). Coulombic attractions between partially chargedmain-chain atoms stabilise the right-handed twist found in most Œ≤-strands. Journal of Molecular Biology, 248(2), 374‚Äì384.</span></a></li>
<li><a href="https://doi.org/10.1186/1472-6807-5-17" target="_blank"><span id="Martin">Martin, J., Letellier, G., Marin, A., Taly, J.-F., de Brevern, A. G., &amp; Gibrat, J.-F. (2005). Protein secondary structure assignment revisited: a detailed analysis of different assignment methods. BMC Structural Biology, 5(1).</span></a></li>
<li><a href="https://doi.org/10.1016/0022-2836(88)90368-3" target="_blank"><span id="Milner-White">Milner-White, E. J., Ross, B. M., Ismail, R., Belhadj-Mostefa, K., &amp; Poet, R. (1988). One type of gamma-turn, rather than the other gives rise to chain-reversal in proteins. Journal of Molecular Biology, 204(3), 777‚Äì782.</span></a></li>
<li><a href="https://doi.org/10.1016/0968-0896(96)00051-x" target="_blank"><span id="Nesloney">Nesloney, C. L., &amp; Kelly, J. W. (1996). Progress towards understanding Œ≤-sheet structure. Bioorganic &amp; Medicinal Chemistry, 4(6), 739‚Äì766.</span></a></li>
<li><a href="https://doi.org/10.1021/acs.chemrev.5b00623" target="_blank"><span id="Papaleo">Papaleo, E., Saladino, G., Lambrughi, M., Lindorff-Larsen, K., Gervasio, F. L., &amp; Nussinov, R. (2016). The role of protein loops and linkers in conformational dynamics and allostery. Chemical Reviews, 116(11), 6391‚Äì6423.</span></a></li>
<li><a href="https://doi.org/10.1002/(sici)1097-0282(199606)38:6%3C705::aid-bip3%3E3.0.co;2-v" target="_blank"><span id="Pavone">Pavone, V., Gaeta, G., Lombardi, A., Nastri, F., Maglio, O., Isernia, C., &amp; Saviano, M. (1996). Discovering protein secondary structures: Classification and description of isolated Œ±-turns. Biopolymers, 38(6), 705‚Äì721.</span></a></li>
<li><a href="https://doi.org/10.1002/pro.5560050515" target="_blank"><span id="Rajashankar">Rajashankar, K. R., &amp; Ramakumar, S. (1996). œÄ‚ÄêTurns in proteins and peptides: Classification, conformation, occurrence, hydration and sequence. Protein Science: A Publication of the Protein Society, 5(5), 932‚Äì946.</span></a></li>
<li><a href="https://doi.org/10.1002/prot.22357" target="_blank"><span id="Ranganathan">Ranganathan, S., Izotov, D., Kraka, E., &amp; Cremer, D. (2009). Description and recognition of regular and distorted secondary structures in proteins using the automated protein structure analysis method. Proteins, 76(2), 418‚Äì438.</span></a></li>
<li><a href="https://doi.org/10.1016/s0022-2836(63)80023-6" target="_blank"><span id="Ramachandran">Ramachandran, G. N., Ramakrishnan, C., &amp; Sasisekharan, V. (1963). Stereochemistry of polypeptide chain configurations. Journal of Molecular Biology, 7(1), 95‚Äì99.</span></a></li>
<li><a href="https://doi.org/10.1073/pnas.75.6.2574" target="_blank"><span id="Richardson">Richardson, J. S., Getzoff, E. D., &amp; Richardson, D. C. (1978). The beta bulge: a common small unit of nonrepetitive protein structure. Proceedings of the National Academy of Sciences of the United States of America, 75(6), 2574‚Äì2578.</span></a></li>
<li><a href="https://doi.org/10.1016/j.jsb.2010.09.001" target="_blank"><span id="Riek">Riek, R. P., &amp; Graham, R. M. (2011). The elusive œÄ-helix. Journal of Structural Biology, 173(1), 153‚Äì160.</span></a></li>
<li><a href="https://doi.org/10.1021/jp002590t" target="_blank"><span id="Shamovsky">Shamovsky, I. L., Ross, G. M., &amp; Riopelle, R. J. (2000). Theoretical Studies on the Origin of Œ≤-sheet Twisting. The Journal of Physical Chemistry. B, 104(47), 11296‚Äì11307.</span></a></li>
<li><a href="https://doi.org/10.1016/s1359-0278(96)00046-6" target="_blank"><span id="Smith">Smith, L. J., Fiebig, K. M., Schwalbe, H., &amp; Dobson, C. M. (1996). The concept of a random coil: Residual structure in peptides and denatured proteins. Folding &amp; Design, 1(5), R95‚ÄìR106.</span></a></li>
<li><a href="https://doi.org/10.1107/s010876818400210x" target="_blank"><span id="Taylor">Taylor, R., Kennard, O., &amp; Versichel, W. (1984). The geometry of the N‚ÄìH...O=C hydrogen bond. 3. Hydrogen-bond distances and angles. Acta Crystallographica. Section B, Structural Science, 40(3), 280‚Äì288.</span></a></li>
<li><a href="https://doi.org/10.1002/chem.201501467" target="_blank"><span id="Toniolo_a">Toniolo, C., Crisma, M., Moretto, A., Peggion, C., Formaggio, F., Alem√°n, C., Cativiela, C., Ramakrishnan, C., &amp; Balaram, P. (2015). Peptide Œ¥‚Äêturn: Literature survey and recent progress. Chemistry (Weinheim an Der Bergstrasse, Germany), 21(40), 13866‚Äì13877.</span></a></li>
<li><a href="https://doi.org/10.1002/bip.22911" target="_blank"><span id="Toniolo_b">Toniolo, C., Crisma, M., Formaggio, F., Alem√°n, C., Ramakrishnan, C., Kalmankar, N., &amp; Balaram, P. (2017). Intramolecular backbone¬∑¬∑¬∑backbone hydrogen bonds in polypeptide conformations. The other way around: …õ‚Äêturn. Biopolymers, 108(1).</span></a></li>
<li><a href="https://doi.org/10.1073/pnas.76.1.19" target="_blank"><span id="Weatherford">Weatherford, D. W., &amp; Salemme, F. R. (1979). Conformations of twisted parallel beta-sheets and the origin of chirality in protein structures. Proceedings of the National Academy of Sciences of the United States of America, 76(1), 19‚Äì23.</span></a></li>
</ol>

<script type="text/javascript" src="./moleculeLoaded.js"></script>
<script type="text/javascript" src="./subviews.js"></script>
<script type="text/javascript" src="./controls.js"></script>
<script type="text/javascript" src="./builder.js"></script>
<script type="text/javascript" src="./ramachandran.js"></script>

<script>

// Handle reference links behavior
// Called when someone clicks reference: <a href="#reference-3" onclick="reference(this.href);">
  function reference(anchor) {
    var spanID = anchor.split("#")[1];
    var span = document.getElementById(spanID);
    blinkOn(span);
  }

  function blinkOn(span) {
    span.counter = 0;
    // Grab computed style so we know the element‚Äôs original text colour:
    span.defColor       = window.getComputedStyle(span).color;
    // Highlight colour; APP brand colour = RGB(128, 30, 101):
    span.highlightColor = "rgb(0, 0, 0)";
    //increase the number if you want the blink to be slower
    span.alertTimerId = setInterval(function() {
      blinkOnce(span.id);
    }, 100);
  }

  function blinkOnce(spanID) {
    var span = document.getElementById(spanID);
    span.counter++;

    // Toggle between original colour and highlight colour.
    // If inline style.color is empty or matches defColour, switch to highlight.
    if (span.style.color === span.defColor || span.style.color === "") {
      span.style.color = span.highlightColor;
    } else {
      span.style.color = span.defColor;
    }

    // After ~20 toggles (‚âà2 second depending on the setinterval), stop and restore original colour: @martin increase the number if you want the blink to last longer
    if (span.counter > 20) {
      blinkOff(span);
    }
  }

  function blinkOff(span) {
    clearInterval(span.alertTimerId);
    span.style.color = span.defColor;
  }


// Handle collapsibles behavior
var coll = document.getElementsByClassName("collapsible");
var i;

for (i = 0; i < coll.length; i++) {
  coll[i].addEventListener("click", function() {
    this.classList.toggle("active");
    var content = this.nextElementSibling;
    var symbol = this.lastChild;
    if (content.style.display === "block") {
      content.style.display = "none";
    } else {
      content.style.display = "block";
    }
    if (symbol.innerHTML === "-") {
      symbol.innerHTML = "+";
    } else {
      symbol.innerHTML = "-";
    }
  });
}

// Handle hoverables behavior
var coll = document.getElementsByClassName("hoverable");
var i;

for (i = 0; i < coll.length; i++) {
  coll[i].addEventListener("mouseover", function() {
    var coll = document.getElementsByClassName("hoverable");
    for (i = 0; i < coll.length; i++) {
	if (coll[i] != this) {
		coll[i].dataset.active = false;
	}
    }
    if (this.dataset.active == "false") {
	    var toggle, residues, chains, currentSelection;
	    var commands = "";
	    switch (this.id) {
		case "helices_text1":
			commands += '!exit;reset;set perspectiveDepth off;zoom 250;';
			commands += 'select all;trace only;trace 0.2;color trace [103, 203, 223];' // Show trace only
			commands += 'select (*.CB or *.CA);wireframe on;wireframe reset;color TRANSLUCENT 1;'; // Show sidechains
			commands += 'color (*.CA) [103, 203, 223];color (*.CB) [122, 115, 191];'; // Color sidechains
			commands += 'select all;hbonds off;' // Hide hydrogen bonds
			commands += 'dipole bonds off;'; // Hide dipoles
			commands += 'script helicesScript.txt;'; // Show axis and axis arrow
			Jmol.script(jmol_helices, commands);
			toggle = document.getElementById("helices_backbone_toggle");
			toggle.innerHTML = "Show backbone atoms";
			toggle.dataset.backbone = "hidden";
			toggle = document.getElementById("helices_sidechains_toggle");
			toggle.innerHTML = "Hide sidechains";
			toggle.dataset.sidechains = "visible";
			toggle = document.getElementById("helices_hbonds_toggle");
			toggle.innerHTML = "Show h-bonds";
			toggle.dataset.hbonds = "hidden";
			toggle = document.getElementById("helices_axis_toggle");
			toggle.innerHTML = "Hide helical axis";
			toggle.dataset.axis = "visible";
			toggle = document.getElementById("helices_termini_toggle");
			toggle.innerHTML = "Show termini &amp; caps";
			toggle.dataset.termini = "hidden";
			toggle = document.getElementById("helices_dipoles_toggle");
			toggle.innerHTML = "Show dipole moments";
			toggle.dataset.termini = "hidden";
			
		break;
		case "helices_text2":
			commands += '!exit;reset;set perspectiveDepth on;zoom 150;rotate y -45;rotate z 35;';
			commands += 'select all;trace only;trace 0.2;color trace [103, 203, 223];' // Show trace only
			commands += 'select (sidechains or *.CA);wireframe off;'; // Hide sidechains
			commands += 'delete $ line1;delete $ line2;'; // Hide axis and axis arrow
			commands += 'select backbone;hbonds calculate;set hbondsBackbone TRUE;hbonds on;color hbonds [247,148,36];hbonds 0.1;'; // Show hydrogen bonds
			commands += 'color trace [103, 203, 223];'; // Color trace
			commands += 'select (2-5);color trace [0, 90, 157];select (19-22);color trace [181, 23, 38];'; // Color termini
			commands += 'select (2);color trace blue;select (22);color trace red;'; // Color caps
			commands += 'dipole bonds off;'; // Hide dipoles
			Jmol.script(jmol_helices, commands);
			toggle = document.getElementById("helices_backbone_toggle");
			toggle.innerHTML = "Show backbone atoms";
			toggle.dataset.backbone = "hidden";
			toggle = document.getElementById("helices_sidechains_toggle");
			toggle.innerHTML = "Show sidechains";
			toggle.dataset.sidechains = "hidden";
			toggle = document.getElementById("helices_hbonds_toggle");
			toggle.innerHTML = "Hide h-bonds";
			toggle.dataset.hbonds = "visible";
			toggle = document.getElementById("helices_axis_toggle");
			toggle.innerHTML = "Show helical axis";
			toggle.dataset.axis = "hidden";
			toggle = document.getElementById("helices_termini_toggle");
			toggle.innerHTML = "Hide termini &amp; caps";
			toggle.dataset.termini = "visible";
			toggle = document.getElementById("helices_dipoles_toggle");
			toggle.innerHTML = "Show dipole moments";
			toggle.dataset.dipoles = "hidden";
		break;
		case "helices_text3":
			commands += '!exit;reset;set perspectiveDepth on;zoom 100;rotate x 90;rotate y 135;';
			commands += 'delete $ line1;delete $ line2;'; // Hide axis and axis arrow
			commands += 'select all;hbonds off;'; // Hide hydrogen bonds
			commands += 'select all;color atoms cpk;wireframe on;wireframe reset;select all;color atoms TRANSLUCENT 0.75;'; // Show backbone
			commands += 'dipole calculate bonds;dipole bonds on;dipole WIDTH 0.2 CROSS OFFSET 1.0 OFFSETSIDE 0.6;set dipoleScale 2;color dipole black;'; // Show dipoles
			commands += 'trace off;'; // Hide trace
			commands += 'select (sidechains or *.CA);wireframe off;'; // Hide sidechains
			
			commands += 'select {atomName=HN and resno=2};label \"<color black>Œ¥<sup>+</sup></color>\";';
			commands += 'select {atomName=O and resno=22};label \"<color black>Œ¥<sup>-</sup></color>\";';
			
			Jmol.script(jmol_helices, commands);
			toggle = document.getElementById("helices_backbone_toggle");
			toggle.innerHTML = "Hide backbone atoms";
			toggle.dataset.backbone = "visible";
			toggle = document.getElementById("helices_sidechains_toggle");
			toggle.innerHTML = "Show sidechains";
			toggle.dataset.sidechains = "hidden";
			toggle = document.getElementById("helices_hbonds_toggle");
			toggle.innerHTML = "Show h-bonds";
			toggle.dataset.hbonds = "hidden";
			toggle = document.getElementById("helices_axis_toggle");
			toggle.innerHTML = "Show helical axis";
			toggle.dataset.axis = "hidden";
			toggle = document.getElementById("helices_termini_toggle");
			toggle.innerHTML = "Show termini &amp; caps";
			toggle.dataset.termini = "hidden";
			toggle = document.getElementById("helices_dipoles_toggle");
			toggle.innerHTML = "Hide dipole moments";
			toggle.dataset.dipoles = "visible";
		break;
		case "strands_text1":
			residues = getValueByKey(moleculeProps, "2TRX")["residues"];
			chains = getValueByKey(moleculeProps, "2TRX")["chains"];
			commands += '!exit;';
			commands += 'select (sidechains or *.CA);wireframe off;'; // Hide sidechains
			currentSelection = "(" + getSelectionString(residues, chains) + ")";
			commands += 'select {' + currentSelection + ' and backbone};hbonds calculate;set hbondsBackbone TRUE;hbonds on;color hbonds [247,148,36];hbonds 0.1;'; // Show hydrogen bonds
			commands += getAlignmentCommands(jmol_strands, residues[2][0], residues[2][1], chains[0][0], chains[0][1]);
			commands += 'delete $ parallel; delete $ antiparallel; color cartoon [103, 203, 223];';
			commands += 'zoom 200;rotate x -30;translate y -12;rotate spin y 5;';
			Jmol.script(jmol_strands, commands);
		break;
		case "strands_text2":
			residues = getValueByKey(moleculeProps, "2TRX")["residues"];
			chains = getValueByKey(moleculeProps, "2TRX")["chains"];
			commands += '!exit;';
			commands += 'select (sidechains or *.CA);wireframe off;'; // Hide sidechains
			commands += 'select all;hbonds off;'; // Hide hydrogen bonds
			commands += getAlignmentCommands(jmol_strands, residues[2][0], residues[2][1], chains[0][0], chains[0][1]);
			commands += 'zoom 200;rotate x -30;translate y -12;rotate spin y 5;';
			commands += 'script strandsScript.txt;'; // Indicate parallel and anti-parallel strands alternatingly
			Jmol.script(jmol_strands, commands);
		break;
		case "strands_text3":
			residues = getValueByKey(moleculeProps, "2TRX")["residues"];
			chains = getValueByKey(moleculeProps, "2TRX")["chains"];
			commands += '!exit;';
			commands += 'select all;hbonds off;'; // Hide hydrogen bonds
			currentSelection = "(" + getSelectionString(residues, chains) + ")";
			commands += 'select {' + currentSelection + ' and (*.CB or *.CA)};color (*.CB) [122, 115, 191];wireframe on;wireframe reset;'; // Show sidechains
			commands += getAlignmentCommands(jmol_strands, residues[2][0], residues[2][1], chains[0][0], chains[0][1]);
			commands += 'delete $ parallel; delete $ antiparallel; color cartoon [103, 203, 223];';
			commands += 'zoom 200;rotate x -30;translate y -12;rotate spin y 5;';
			Jmol.script(jmol_strands, commands);
		break;
		case "righttwist_text1":
			residues = getValueByKey(moleculeProps, "1BA7")["residues"];
			chains = getValueByKey(moleculeProps, "1BA7")["chains"];
			// Create axes illustrating twist sense
			commands += 'delete $ line2;delete $ line3;delete $ arc2;delete $ arc3;'; // Hide other axes and axis arrows
			// Start of strand
			commands += 'xyzstart = {' + residues[0][0] + ':' + chains[1][0] + '.CA}.xyz;';
			// End of strand
			commands += 'xyzend = {' + residues[0][1] + ':' + chains[1][1] + '.CA}.xyz;';
			commands += 'draw line1 ARROW width 0.25 color [233, 193, 85] @xyzstart @xyzend;';
			commands += 'draw arc1 ARROW ARC SCALE 4.0 DIAMETER 0.1 color [233, 193, 85] @xyzstart @xyzend {0, 0, 0} {90 270 0.5};';
			Jmol.script(jmol_righttwist, commands);
		break;
		case "righttwist_text2":
			residues = getValueByKey(moleculeProps, "1BA7")["residues"];
			chains = getValueByKey(moleculeProps, "1BA7")["chains"];
			// Create axes illustrating twist sense
			commands += 'delete $ line1;delete $ line3;delete $ arc1;delete $ arc3;'; // Hide other axes and axis arrows
			// Start of sheet
			commands += 'xyzstart = ({' + residues[0][0] + ':' + chains[0][0] + '.CA}.xyz + {' + residues[1][1] + ':' + chains[1][1] + '.CA}.xyz) / 2;';
			// End of sheet
			commands += 'xyzend = ({' + residues[1][0] + ':' + chains[1][0] + '.CA}.xyz + {' + residues[0][1] + ':' + chains[0][1] + '.CA}.xyz) / 2;';
			commands += 'draw line2 ARROW width 0.25 color [33, 43, 56] @xyzstart @xyzend;';
			commands += 'draw arc2 ARROW ARC SCALE 10.0 DIAMETER 0.1 color [33, 43, 56] @xyzstart @xyzend {0, 0, 0} {90 270 0.5};';
			Jmol.script(jmol_righttwist, commands);
		break;
		case "righttwist_text3":
			residues = getValueByKey(moleculeProps, "1BA7")["residues"];
			chains = getValueByKey(moleculeProps, "1BA7")["chains"];
			// Create axes illustrating twist sense
			commands += 'delete $ line1;delete $ line2;delete $ arc1;delete $ arc2;'; // Hide other axes and axis arrows
			// Start of sheet (orthogonal)
			commands += 'xyzstart = ({' + residues[1][0] + ':' + chains[1][0] + '.CA}.xyz + {' + residues[1][1] + ':' + chains[1][1] + '.CA}.xyz) / 2;';
			// End of sheet (orthogonal)
			commands += 'xyzend = ({' + residues[0][0] + ':' + chains[0][0] + '.CA}.xyz + {' + residues[0][1] + ':' + chains[0][1] + '.CA}.xyz) / 2;';
			commands += 'draw line3 ARROW width 0.25 color [122, 115, 191] @xyzstart @xyzend;';
			commands += 'draw arc3 ARROW ARC SCALE 10.0 DIAMETER 0.1 color [122, 115, 191] @xyzend @xyzstart {0, 0, 0} {0 270 0.5};';
			Jmol.script(jmol_righttwist, commands);
		break;
	    }
	    this.dataset.active = true;
    }
  });
}
</script>

<script>
// Handle selection of secondary structure element in overview
var coll = document.getElementsByClassName("clickeable");
var i;

for (i = 0; i < coll.length; i++) {
  coll[i].addEventListener("click", function(evt) {
	evt.stopPropagation();
	var divID = evt.target.id;
	// First, hide all content
	var coll = document.getElementsByClassName("sscontent");
	var i;
	for (i = 0; i < coll.length; i++) {
		if (coll[i].id != divID) {
			coll[i].style.display = 'none';
		}
	}
	if (document.getElementById(divID + "_content")) {
		if (divID != 'mainBox') {
			document.getElementById(divID + "_content").style.display = 'flex';
			if (contentLoaded[divID] == false) {
				let residues, chanis;
				switch(divID) {
					case "helices":
						$('#JmolDiv_helices').html( Jmol.getAppletHtml(jmol_helices, JmolInfo) );
						Jmol.script(jmol_helices, 'set disablePopupMenu TRUE;frank off;load idealalpha.pdb;trace only;trace 0.2;hide (*.HA);set perspectiveDepth off;zoom 250;select (*.CB or *.CA);color (*.CB) [122, 115, 191];wireframe on;wireframe reset;color [103, 203, 223];color (*.CB) [122, 115, 191];script helicesScript.txt;');
					break;
					case "strands":
						residues = getValueByKey(moleculeProps, "2TRX")["residues"];
						chains = getValueByKey(moleculeProps, "2TRX")["chains"];
						var currentSelection = "(" + getSelectionString(residues, chains) + ")";
						$('#JmolDiv_strands').html( Jmol.getAppletHtml(jmol_strands, JmolInfo) );
						Jmol.script(jmol_strands, 'set disablePopupMenu TRUE;frank off;load 2trx.pdb;cartoon only;set cartoonFancy on;color [103, 203, 223];select {' + currentSelection + ' and (*.CB or *.CA)};color (*.CB) [122, 115, 191];wireframe on;wireframe reset;select {' + currentSelection + ' and backbone};hbonds calculate;set hbondsBackbone TRUE;hbonds on;color hbonds [247,148,36];hbonds 0.1;');
					break;
					case "turns":
						$('#JmolDiv_turns').html( Jmol.getAppletHtml(jmol_turns, JmolInfo) );
						Jmol.script(jmol_turns, 'set disablePopupMenu TRUE;frank off;load 1arb.pdb;cartoon only;set cartoonFancy on;color [103, 203, 223];');
					break;
					case "regular":
						$('#JmolDiv_regular').html( Jmol.getAppletHtml(jmol_regular, JmolInfo) );
						Jmol.script(jmol_regular, 'set disablePopupMenu TRUE;frank off;set perspectiveDepth off;load 6ec0.pdb;hide chain=B;cartoon only;set cartoonFancy on;color [103, 203, 223];');
					break;
					case "distorted":
						$('#JmolDiv_distorted').html( Jmol.getAppletHtml(jmol_distorted, JmolInfo) );
						distorted(0);
					break;
					case "threeten":
						$('#JmolDiv_310').html( Jmol.getAppletHtml(jmol_310, JmolInfo) );
						Jmol.script(jmol_310, 'set disablePopupMenu TRUE;frank off;set perspectiveDepth off;load 1gpb.pdb;cartoon only;set cartoonFancy on;color [103, 203, 223];');
					break;
					case "pihelix":
						$('#JmolDiv_pi').html( Jmol.getAppletHtml(jmol_pi, JmolInfo) );
						Jmol.script(jmol_pi, 'set disablePopupMenu TRUE;frank off;set perspectiveDepth off;load 2qd1.pdb;cartoon only;set cartoonFancy on;color [103, 203, 223];');
					break;
					case "parallel":
						$('#JmolDiv_parallel').html( Jmol.getAppletHtml(jmol_parallel, JmolInfo) );
						Jmol.script(jmol_parallel, 'set disablePopupMenu FALSE;frank off;set perspectiveDepth off;load 2beg.pdb;cartoon only;set cartoonFancy on;color [103, 203, 223];');
					break;
					case "lefttwist":
						$('#JmolDiv_lefttwist').html( Jmol.getAppletHtml(jmol_lefttwist, JmolInfo) );
						Jmol.script(jmol_lefttwist, 'set disablePopupMenu TRUE;frank off;set perspectiveDepth off;load 1ofs.pdb;cartoon only;set cartoonFancy on;color [103, 203, 223];');
					break;
					case "relaxed":
						$('#JmolDiv_relaxed').html( Jmol.getAppletHtml(jmol_relaxed, JmolInfo) );
						Jmol.script(jmol_relaxed, 'set disablePopupMenu TRUE;frank off;set perspectiveDepth off;load 2yxf.pdb;cartoon only;set cartoonFancy on;color [103, 203, 223];');
					break;
					case "righttwist":
						$('#JmolDiv_righttwist').html( Jmol.getAppletHtml(jmol_righttwist, JmolInfo) );
						Jmol.script(jmol_righttwist, 'set disablePopupMenu TRUE;frank off;set perspectiveDepth off;load 1ba7.pdb;cartoon only;set cartoonFancy on;color [103, 203, 223];');
					break;
					case "bulge":
						$('#JmolDiv_bulge').html( Jmol.getAppletHtml(jmol_bulge, JmolInfo) );
						bulge(0);
					break;
					case "piturns":
						$('#JmolDiv_piturn').html( Jmol.getAppletHtml(jmol_piturn, JmolInfo) );
						turn(0, 'piturn');
					break;
					case "alphaturns":
						$('#JmolDiv_alphaturn').html( Jmol.getAppletHtml(jmol_alphaturn, JmolInfo) );
						turn(0, 'alphaturn');
					break;
					case "betaturns":
						$('#JmolDiv_betaturn').html( Jmol.getAppletHtml(jmol_betaturn, JmolInfo) );
						turn(0, 'betaturn');
					break;
					case "gammaturns":
						$('#JmolDiv_gammaturn').html( Jmol.getAppletHtml(jmol_gammaturn, JmolInfo) );
						turn(0, 'gammaturn');
					break;
					case "epsilonturns":
						$('#JmolDiv_epsilonturn').html( Jmol.getAppletHtml(jmol_epsilonturn, JmolInfo) );
						turn(0, 'epsilonturn');
					break;
					case "deltaturns":
						$('#JmolDiv_deltaturn').html( Jmol.getAppletHtml(jmol_deltaturn, JmolInfo) );
						turn(0, 'deltaturn');
					break;
					case "coil":
						$('#JmolDiv_coil').html( Jmol.getAppletHtml(jmol_coil, JmolInfo) );
						Jmol.script(jmol_coil, 'set disablePopupMenu TRUE;frank off;set perspectiveDepth off;load 1lri.pdb;hide chain=B;cartoon only;set cartoonFancy on;color [103, 203, 223];');
					break;
				}
				contentLoaded[divID] = true;
			} else {
				switch(divID) {
					case "helices":
						setView('idealalpha', '0');
					break;
					case "strands":
						setView('2TRX', '0');
					break;
					case "turns":
						setView('1ARB', '0');
					break;
					case "pihelix":
						setView('2QD1', '1');
					break;
					case "threeten":
						setView('1GPB', '1');
					break;
				}
			}
			if (divID == "distorted") {
				distorted(0); // Jmol applet is empty if this is not called again
			}
		}
	}
  });
}
</script>

<!-- To make the parent page do the scrolling -->

<script>
	document.addEventListener('click', function(evt) {

	  const link = evt.target.closest('a[href^="#"]');
	  if (!link) return;

	  const targetID = link.getAttribute('href').substring(1);
	  const targetElem = document.getElementById(targetID);
	  if (!targetElem) return;

	  // Distance from top of iframe‚Äôs document to the anchor
	  const offsetY = targetElem.getBoundingClientRect().top + window.pageYOffset;

	  // Send the offset up to the parent page
	  parent.postMessage({
		type: 'scrollToReference',
		id: targetID,
		y: offsetY
	  }, '*');

	  // ( @Martin: do NOT call preventDefault here, so the widget‚Äôs own ‚Äúhash‚Äù still updates
	  // and any :target‚Äêbased highlight will still work inside the iframe.)
	});
  </script>
  
<script>
	$(window).on("load", function() {
		// Hide loading image(s)
		var divsToHide = document.getElementsByClassName("loading");
	    for (var i = 0; i < divsToHide.length; i++) {
	        divsToHide[i].style.visibility = "hidden";
	    //    //divsToHide[i].style.display = "none";
	    }
		console.log("Loading image hidden...");
	});
</script>
	


<div id="_Loader-status" style="position: absolute; bottom: 4px; left: 4px; padding: 2px 8px; z-index: 109007; background-color: rgb(142, 0, 0); color: yellow; font-family: Arial, sans-serif; font-size: 10pt; white-space: nowrap; opacity: 0.05; display: none;">Application loaded.</div></body><iframe style="display:none" width="0" height="0" frameborder="0" tabindex="-1" src="./JmolTracker.html"></iframe></html>
